

<!DOCTYPE html>


<html lang="en" data-content_root="" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

    <title>R stack for -omics data &#8212; Bioinformatic cheatsheet</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "";
  </script>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="_static/styles/theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="_static/styles/bootstrap.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="_static/styles/pydata-sphinx-theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />

  
  <link href="_static/vendor/fontawesome/6.5.2/css/all.min.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
  <link rel="preload" as="font" type="font/woff2" crossorigin href="_static/vendor/fontawesome/6.5.2/webfonts/fa-solid-900.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="_static/vendor/fontawesome/6.5.2/webfonts/fa-brands-400.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="_static/vendor/fontawesome/6.5.2/webfonts/fa-regular-400.woff2" />

    <link rel="stylesheet" type="text/css" href="_static/pygments.css" />
    <link rel="stylesheet" type="text/css" href="_static/styles/sphinx-book-theme.css" />
    <link rel="stylesheet" type="text/css" href="_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="_static/mystnb.4510f1fc1dee50b3e5859aac5469c37c29e427902b24a333a5f9fcb2f0b3ac41.css" />
    <link rel="stylesheet" type="text/css" href="_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="_static/sphinx-design.4cbf315f70debaebd550c87a6162cf0f.min.css" />
  
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b" />
<link rel="preload" as="script" href="_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b" />
  <script src="_static/vendor/fontawesome/6.5.2/js/all.min.js?digest=dfe6caa3a7d634c4db9b"></script>

    <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
    <script src="_static/jquery.js"></script>
    <script src="_static/underscore.js"></script>
    <script src="_static/_sphinx_javascript_frameworks_compat.js"></script>
    <script src="_static/doctools.js"></script>
    <script src="_static/sphinx_highlight.js"></script>
    <script src="_static/clipboard.min.js"></script>
    <script src="_static/copybutton.js"></script>
    <script src="_static/scripts/sphinx-book-theme.js"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="_static/togglebutton.js"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script src="_static/design-tabs.js"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script async="async" src="_static/sphinx-thebe.js"></script>
    <script>DOCUMENTATION_OPTIONS.pagename = 'R_stack_cheatsheet';</script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="System setup" href="setup.html" />
    <link rel="prev" title="Git &amp; GitHub" href="git.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <div id="pst-skip-link" class="skip-link d-print-none"><a href="#main-content">Skip to main content</a></div>
  
  <div id="pst-scroll-pixel-helper"></div>
  
  <button type="button" class="btn rounded-pill" id="pst-back-to-top">
    <i class="fa-solid fa-arrow-up"></i>Back to top</button>

  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-primary-sidebar-checkbox"/>
  <label class="overlay overlay-primary" for="pst-primary-sidebar-checkbox"></label>
  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-secondary-sidebar-checkbox"/>
  <label class="overlay overlay-secondary" for="pst-secondary-sidebar-checkbox"></label>
  
  <div class="search-button__wrapper">
    <div class="search-button__overlay"></div>
    <div class="search-button__search-container">
<form class="bd-search d-flex align-items-center"
      action="search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         id="search-input"
         placeholder="Search this book..."
         aria-label="Search this book..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form></div>
  </div>

  <div class="pst-async-banner-revealer d-none">
  <aside id="bd-header-version-warning" class="d-none d-print-none" aria-label="Version warning"></aside>
</div>

  
    <header class="bd-header navbar navbar-expand-lg bd-navbar d-print-none">
    </header>
  

  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      
      
      <div class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
    
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">

  
    
  

<a class="navbar-brand logo" href="intro.html">
  
  
  
  
  
    
    
      
    
    
    <img src="_static/logo.png" class="logo__image only-light" alt="Bioinformatic cheatsheet - Home"/>
    <script>document.write(`<img src="_static/logo.png" class="logo__image only-dark" alt="Bioinformatic cheatsheet - Home"/>`);</script>
  
  
</a></div>
        <div class="sidebar-primary-item">

 <script>
 document.write(`
   <button class="btn search-button-field search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass"></i>
    <span class="search-button__default-text">Search</span>
    <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd class="kbd-shortcut__modifier">K</kbd></span>
   </button>
 `);
 </script></div>
        <div class="sidebar-primary-item"><nav class="bd-links bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item navbar-nav active">
        
        <ul class="nav bd-sidenav bd-sidenav__home-link">
            <li class="toctree-l1">
                <a class="reference internal" href="intro.html">
                    Bioinformatics cheatsheets
                </a>
            </li>
        </ul>
        <ul class="current nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="conda_cheatsheet.html">Conda</a></li>
<li class="toctree-l1"><a class="reference internal" href="git.html">Git &amp; GitHub</a></li>
<li class="toctree-l1 current active"><a class="current reference internal" href="#">R stack for -omics data</a></li>



<li class="toctree-l1"><a class="reference internal" href="setup.html">System setup</a></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="bfb.html">Course notes: Bioinformatics for Biologists</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="Week1.html">Week 1 - setup, sequencing QC, mapping, variant calling</a></li>
<li class="toctree-l2"><a class="reference internal" href="Week2-workflows.html">Week 2 - workflows with Nextflow</a></li>
</ul>
</details></li>
<li class="toctree-l1"><a class="reference internal" href="jupyterlabtrick_wsl.html">WSL incompatibility solutions</a></li>
</ul>

    </div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main" role="main">
        
        

<div class="sbt-scroll-pixel-helper"></div>

          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article d-print-none">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"><button class="sidebar-toggle primary-toggle btn btn-sm" title="Toggle primary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
  <span class="fa-solid fa-bars"></span>
</button></div>
      
    </div>
  
  
    <div class="header-article-items__end">
      
        <div class="header-article-item">

<div class="article-header-buttons">





<div class="dropdown dropdown-source-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Source repositories">
    <i class="fab fa-github"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="https://github.com/executablebooks/jupyter-book" target="_blank"
   class="btn btn-sm btn-source-repository-button dropdown-item"
   title="Source repository"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fab fa-github"></i>
  </span>
<span class="btn__text-container">Repository</span>
</a>
</li>
      
      
      
      
      <li><a href="https://github.com/executablebooks/jupyter-book/issues/new?title=Issue%20on%20page%20%2FR_stack_cheatsheet.html&body=Your%20issue%20content%20here." target="_blank"
   class="btn btn-sm btn-source-issues-button dropdown-item"
   title="Open an issue"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-lightbulb"></i>
  </span>
<span class="btn__text-container">Open issue</span>
</a>
</li>
      
  </ul>
</div>






<div class="dropdown dropdown-download-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Download this page">
    <i class="fas fa-download"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="_sources/R_stack_cheatsheet.md" target="_blank"
   class="btn btn-sm btn-download-source-button dropdown-item"
   title="Download source file"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="btn__text-container">.md</span>
</a>
</li>
      
      
      
      
      <li>
<button onclick="window.print()"
  class="btn btn-sm btn-download-pdf-button dropdown-item"
  title="Print to PDF"
  data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="btn__text-container">.pdf</span>
</button>
</li>
      
  </ul>
</div>




<button onclick="toggleFullScreen()"
  class="btn btn-sm btn-fullscreen-button"
  title="Fullscreen mode"
  data-bs-placement="bottom" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>



<script>
document.write(`
  <button class="btn btn-sm nav-link pst-navbar-icon theme-switch-button" title="light/dark" aria-label="light/dark" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="theme-switch fa-solid fa-sun fa-lg" data-mode="light"></i>
    <i class="theme-switch fa-solid fa-moon fa-lg" data-mode="dark"></i>
    <i class="theme-switch fa-solid fa-circle-half-stroke fa-lg" data-mode="auto"></i>
  </button>
`);
</script>


<script>
document.write(`
  <button class="btn btn-sm pst-navbar-icon search-button search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass fa-lg"></i>
  </button>
`);
</script>
<button class="sidebar-toggle secondary-toggle btn btn-sm" title="Toggle secondary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="fa-solid fa-list"></span>
</button>
</div></div>
      
    </div>
  
</div>
</div>
              
              

<div id="jb-print-docs-body" class="onlyprint">
    <h1>R stack for -omics data</h1>
    <!-- Table of contents -->
    <div id="print-main-content">
        <div id="jb-print-toc">
            
            <div>
                <h2> Contents </h2>
            </div>
            <nav aria-label="Page">
                <ul class="visible nav section-nav flex-column">
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#">R stack for -omics data</a></li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#scope">Scope</a><ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#affy">1. affy</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#limma">2. limma</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#clusterprofiler">3. clusterProfiler</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#pheatmap">4. pheatmap</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#pcatools">5. PCAtools</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#eulerr">6. eulerr</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#corrplot">7. corrplot</a></li>
</ul>
</li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#session-info">Session info</a></li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#more-sources">More sources</a></li>
</ul>

            </nav>
        </div>
    </div>
</div>

              
                
<div id="searchbox"></div>
                <article class="bd-article">
                  
  <section class="tex2jax_ignore mathjax_ignore" id="r-stack-for-omics-data">
<h1>R stack for -omics data<a class="headerlink" href="#r-stack-for-omics-data" title="Permalink to this heading">#</a></h1>
<p>Martyna Muszczek
2024-02-14</p>
</section>
<section class="tex2jax_ignore mathjax_ignore" id="scope">
<h1>Scope<a class="headerlink" href="#scope" title="Permalink to this heading">#</a></h1>
<p>This is a quick overview of my current R stack that I use for proteomic
data analysis. The aim of this cheatsheet is to organize packages and
their standard use as it is easy to get lost in all available
Bioconductor packages that in the end don’t work well or are missing a
lot of functions and need to be combined with another packages. This is
what has worked for me so far and I wanted a place where everything is
summed up in a somewhat sequential order of analysis.</p>
<p>So far I have been mainly working with two or three experimental groups
but there was an instance I had to create long functions just to
automatize DE, term enrichment and their visualization for multiple
groups. These functions can be found under “3.4 Functions for multiple
groups”.</p>
<section id="affy">
<h2>1. affy<a class="headerlink" href="#affy" title="Permalink to this heading">#</a></h2>
<p>Currently used for testing normalization performance with MA plots. Blue
line represents theoretical perfect loess curve, red line - actual loess
curve. The more similar they are, the better.</p>
<p><b>Note:</b> Normalization principle is that majority of features
(proteins/genes) do not change their expression under most conditions
and so their trend is used to normalize samples.</p>
<p>M = log2(x/y) aka log2FC</p>
<p>A = (log2(x) + log2(y))/2 = log2(xy)*1/2, aka means of means=average
expression</p>
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">library</span><span class="p">(</span><span class="n">affy</span><span class="p">)</span>
<span class="c1"># Named list with column indexes for each experimental group</span>
<span class="n">groups</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">list</span><span class="p">(</span><span class="m">1</span><span class="o">:</span><span class="m">3</span><span class="p">,</span><span class="m">4</span><span class="o">:</span><span class="m">6</span><span class="p">,</span><span class="m">7</span><span class="o">:</span><span class="m">9</span><span class="p">)</span>
<span class="nf">names</span><span class="p">(</span><span class="n">groups</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">colnames</span><span class="p">(</span><span class="n">design</span><span class="p">)</span><span class="w"> </span><span class="c1">#from limma pkg - search below</span>

<span class="c1">#take 2 experimental groups from matrix</span>
<span class="n">x</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">rowMeans</span><span class="p">(</span><span class="n">qnorm</span><span class="p">[,</span><span class="nf">c</span><span class="p">(</span><span class="n">groups</span><span class="o">$</span><span class="n">x3D</span><span class="p">)])</span>
<span class="n">y</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">rowMeans</span><span class="p">(</span><span class="n">qnorm</span><span class="p">[,</span><span class="nf">c</span><span class="p">(</span><span class="n">groups</span><span class="o">$</span><span class="n">x2D</span><span class="p">)])</span>

<span class="n">M</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">x</span><span class="w"> </span><span class="o">-</span><span class="w"> </span><span class="n">y</span>
<span class="n">A</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="n">y</span><span class="p">)</span><span class="o">/</span><span class="m">2</span>

<span class="c1">#plot and save</span>
<span class="n">maplot</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">ma.plot</span><span class="p">(</span><span class="n">A</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">A</span><span class="p">,</span><span class="w"> </span><span class="n">M</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">M</span><span class="p">,</span><span class="w"> </span><span class="n">cex</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">)</span>
<span class="nf">title</span><span class="p">(</span><span class="s">&quot;... normalisation&quot;</span><span class="p">)</span>
</pre></div>
</div>
</section>
<section id="limma">
<h2>2. limma<a class="headerlink" href="#limma" title="Permalink to this heading">#</a></h2>
<p>Originally dedicated to microarray and RNA-seq data, which is also
useful for proteomics. Currently used for normalization methods and
differential expression analysis.</p>
<ol class="arabic">
<li><p>Load required packages</p>
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">library</span><span class="p">(</span><span class="n">tidyverse</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">limma</span><span class="p">)</span>
</pre></div>
</div>
</li>
<li><p>Prepare an expression matrix - log2 transformed, normalized, tidy.</p>
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">mat</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">read.delim</span><span class="p">(</span><span class="s">&quot;expr_matrix.tsv&quot;</span><span class="p">)</span>
<span class="n">mat</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">as.matrix</span><span class="p">(</span><span class="n">mat</span><span class="p">)</span>
<span class="nf">head</span><span class="p">(</span><span class="n">mat</span><span class="p">)</span>

<span class="c1">#different normalization methods, assess with boxplots with jitter and MA plots</span>
<span class="c1"># normalizeVSN()</span>
<span class="c1"># normalizeQuantiles()</span>
<span class="c1"># normalizeCyclicLoess()</span>
<span class="c1"># normalizeMedianValues()</span>
</pre></div>
</div>
</li>
<li><p>Create a design table, which is a 0/1 matrix of assigning samples to
an experimental group. There are many ways to do it, but I find the
following one the easiest. In the <code>model.matrix()</code>
function in <code>rep()</code> first digit represents the column
index that will be filled with nr of replicates in a group indicated
by second digit.</p>
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">design</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">model.matrix</span><span class="p">(</span><span class="o">~</span><span class="w"> </span><span class="m">0</span><span class="o">+</span><span class="nf">factor</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="nf">rep</span><span class="p">(</span><span class="m">1</span><span class="p">,</span><span class="m">3</span><span class="p">),</span>
<span class="w">                                    </span><span class="nf">rep</span><span class="p">(</span><span class="m">2</span><span class="p">,</span><span class="m">3</span><span class="p">),</span>
<span class="w">                                    </span><span class="nf">rep</span><span class="p">(</span><span class="m">3</span><span class="p">,</span><span class="m">3</span><span class="p">),</span>
<span class="w">                                    </span><span class="nf">rep</span><span class="p">(</span><span class="m">4</span><span class="p">,</span><span class="m">3</span><span class="p">))))</span>
<span class="c1"># assign names of experimental groups</span>
<span class="nf">colnames</span><span class="p">(</span><span class="n">design</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;ctrl&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;treat1&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;treat2&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;treat3&quot;</span><span class="p">)</span>
<span class="c1"># assign sample names from expression matrix to exp groups or manually</span>
<span class="nf">row.names</span><span class="p">(</span><span class="n">design</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">colnames</span><span class="p">(</span><span class="n">mat</span><span class="p">)</span>
<span class="n">design</span>
</pre></div>
</div>
</li>
<li><p>Create an lm fit and matrix of contrasts - meaning comparisons
between groups that will be made. To make things clear in next
steps, assign a number in the comment as it will be needed for coef
arguments later.</p>
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">fit</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">lmFit</span><span class="p">(</span><span class="n">mat</span><span class="p">,</span><span class="n">design</span><span class="p">)</span>
<span class="n">cont.matrix</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">makeContrasts</span><span class="p">(</span><span class="n">ctrl_treat1</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">ctrl</span><span class="w"> </span><span class="o">-</span><span class="w"> </span><span class="n">treat1</span><span class="p">,</span><span class="w"> </span><span class="c1">#1</span>
<span class="w">                             </span><span class="n">ctrl_treat2</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">ctrl</span><span class="w"> </span><span class="o">-</span><span class="w"> </span><span class="n">treat2</span><span class="p">,</span><span class="w"> </span><span class="c1">#2</span>
<span class="w">                             </span><span class="n">ctrl_treat3</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">ctrl</span><span class="w"> </span><span class="o">-</span><span class="w"> </span><span class="n">treat3</span><span class="p">,</span><span class="w"> </span><span class="c1">#3</span>
<span class="w">                             </span><span class="n">levels</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">design</span><span class="p">)</span><span class="w"> </span>
</pre></div>
</div>
</li>
<li><p>Fit the contrasts and calculate Bayes statistics.</p>
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">fit_cont</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">contrasts.fit</span><span class="p">(</span><span class="n">fit</span><span class="p">,</span><span class="w"> </span><span class="n">cont.matrix</span><span class="p">)</span>
<span class="n">fit_bayes</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">eBayes</span><span class="p">(</span><span class="n">fit_cont</span><span class="p">)</span>
</pre></div>
</div>
</li>
<li><p>Write down the whole differential expression matrix into a
data.frame. ‘coef’ argument indicates which comparison is taken into
account. See that expression matrix has to have protein/gene ids in
rownames.</p>
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">DEP</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">topTable</span><span class="p">(</span><span class="n">fit_bayes</span><span class="p">,</span><span class="w"> </span>
<span class="w">                </span><span class="n">coef</span><span class="o">=</span><span class="m">1</span><span class="p">,</span><span class="w"> </span>
<span class="w">                </span><span class="n">adjust.method</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;BH&quot;</span><span class="p">,</span><span class="w"> </span>
<span class="w">                </span><span class="n">genelist</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">rownames</span><span class="p">(</span><span class="n">mat</span><span class="p">),</span><span class="w"> </span>
<span class="w">                </span><span class="n">resort.by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;logFC&quot;</span><span class="p">,</span><span class="w"> </span>
<span class="w">                </span><span class="n">number</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">nrow</span><span class="p">(</span><span class="n">mat</span><span class="p">))</span>
</pre></div>
</div>
</li>
<li><p>For a quick volcano plot do this:</p>
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">volcanoplot</span><span class="p">(</span><span class="n">fit_bayes</span><span class="p">,</span>
<span class="w">            </span><span class="n">coef</span><span class="o">=</span><span class="m">1</span><span class="p">,</span>
<span class="w">            </span><span class="n">highlight</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">10</span><span class="p">,</span>
<span class="w">            </span><span class="n">names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">rownames</span><span class="p">(</span><span class="n">mat</span><span class="p">))</span>
<span class="nf">abline</span><span class="p">(</span><span class="n">h</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1.2</span><span class="p">,</span><span class="w"> </span><span class="n">v</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">-1.5</span><span class="p">,</span><span class="m">1.5</span><span class="p">),</span><span class="w"> </span><span class="n">lty</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">2</span><span class="p">)</span>
</pre></div>
</div>
</li>
<li><p>A much more preferred version:</p>
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">deps</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">topTable</span><span class="p">(</span><span class="n">fit_bayes</span><span class="p">,</span><span class="w"> </span><span class="n">coef</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="n">adjust</span><span class="o">=</span><span class="s">&quot;BH&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">genelist</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">rownames</span><span class="p">(</span><span class="n">mat</span><span class="p">),</span><span class="w"> </span><span class="n">resort.by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;logFC&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">number</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">nrow</span><span class="p">(</span><span class="n">qnorm</span><span class="p">))</span>

<span class="c1">#Retain columns for volcano plot and assign Category of expression</span>
<span class="n">volc_data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">deps</span>
<span class="n">volc_data</span><span class="o">$</span><span class="n">Category</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&quot;Not significant&quot;</span>
<span class="n">volc_data</span><span class="o">$</span><span class="n">Category</span><span class="p">[</span><span class="n">volc_data</span><span class="o">$</span><span class="n">logFC</span><span class="w"> </span><span class="o">&lt;=</span><span class="w"> </span><span class="m">-1.3</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="n">volc_data</span><span class="o">$</span><span class="n">adj.P.Val</span><span class="w"> </span><span class="o">&lt;=</span><span class="w"> </span><span class="m">0.05</span><span class="p">]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&quot;Down-regulated&quot;</span>
<span class="n">volc_data</span><span class="o">$</span><span class="n">Category</span><span class="p">[</span><span class="n">volc_data</span><span class="o">$</span><span class="n">logFC</span><span class="w"> </span><span class="o">&gt;=</span><span class="w"> </span><span class="m">1.3</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="n">volc_data</span><span class="o">$</span><span class="n">adj.P.Val</span><span class="w"> </span><span class="o">&lt;=</span><span class="w"> </span><span class="m">0.05</span><span class="p">]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&quot;Up-regulated&quot;</span>

<span class="c1">#Assign label of DEPs</span>
<span class="n">volc_data</span><span class="o">$</span><span class="n">Label</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="kc">NA</span>
<span class="n">volc_data</span><span class="o">$</span><span class="n">Label</span><span class="p">[</span><span class="n">volc_data</span><span class="o">$</span><span class="n">Category</span><span class="w"> </span><span class="o">!=</span><span class="w"> </span><span class="s">&quot;Not significant&quot;</span><span class="p">]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">volc_data</span><span class="o">$</span><span class="n">ID</span><span class="p">[</span><span class="n">volc_data</span><span class="o">$</span><span class="n">Category</span><span class="w"> </span><span class="o">!=</span><span class="w"> </span><span class="s">&quot;Not significant&quot;</span><span class="p">]</span>

<span class="c1">#Plot</span>
<span class="nf">theme_set</span><span class="p">(</span><span class="nf">theme_light</span><span class="p">())</span>
<span class="n">v</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">ggplot</span><span class="p">(</span><span class="n">volc_data</span><span class="p">,</span>
<span class="w">            </span><span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">logFC</span><span class="p">,</span>
<span class="w">                </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="p">(</span><span class="m">-1</span><span class="o">*</span><span class="nf">log10</span><span class="p">(</span><span class="n">adj.P.Val</span><span class="p">)),</span>
<span class="w">                </span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Category</span><span class="p">,</span>
<span class="w">                </span><span class="n">label</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Label</span><span class="p">))</span><span class="o">+</span>
<span class="w">  </span><span class="nf">geom_point</span><span class="p">(</span><span class="n">alpha</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">.</span><span class="m">5</span><span class="p">)</span><span class="o">+</span>
<span class="w">  </span><span class="nf">scale_color_manual</span><span class="p">(</span><span class="n">values</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;Down-regulated&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;blue&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;Not significant&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;gray&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;Up-regulated&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;red&quot;</span><span class="p">))</span><span class="o">+</span>
<span class="w">  </span><span class="nf">labs</span><span class="p">(</span><span class="n">title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">names</span><span class="p">(</span><span class="n">coefs</span><span class="p">[</span><span class="n">i</span><span class="p">]),</span>
<span class="w">       </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;-log10 adj. p-value&quot;</span><span class="p">,</span>
<span class="w">       </span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;log2 fold change&quot;</span><span class="p">)</span><span class="o">+</span>
<span class="w">  </span><span class="nf">geom_hline</span><span class="p">(</span><span class="n">yintercept</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1.3</span><span class="p">,</span><span class="w"> </span><span class="n">lty</span><span class="o">=</span><span class="m">2</span><span class="p">)</span><span class="o">+</span>
<span class="w">  </span><span class="nf">geom_vline</span><span class="p">(</span><span class="n">xintercept</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">-1.3</span><span class="p">,</span><span class="m">1.3</span><span class="p">),</span><span class="w"> </span><span class="n">lty</span><span class="o">=</span><span class="m">2</span><span class="p">)</span><span class="o">+</span>
<span class="w">  </span><span class="nf">scale_x_continuous</span><span class="p">(</span><span class="n">limits</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">-5.5</span><span class="p">,</span><span class="m">5.5</span><span class="p">))</span><span class="o">+</span>
<span class="w">  </span><span class="nf">scale_y_continuous</span><span class="p">(</span><span class="n">limits</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">10.5</span><span class="p">))</span>

<span class="c1">#Additional formatting, adding n of upregs and downregs</span>
<span class="n">v</span><span class="o">+</span>
<span class="w">  </span><span class="nf">theme</span><span class="p">(</span><span class="n">legend.position</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;top&quot;</span><span class="p">,</span>
<span class="w">        </span><span class="n">text</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="o">=</span><span class="m">15</span><span class="p">))</span><span class="o">+</span>
<span class="w">  </span><span class="nf">geom_text_repel</span><span class="p">(</span><span class="n">colour</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;black&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">max.overlaps</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">15</span><span class="p">,</span><span class="w"> </span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">3.5</span><span class="p">)</span><span class="o">+</span>
<span class="w">  </span><span class="nf">annotate</span><span class="p">(</span><span class="s">&quot;text&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">-5.5</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="o">=</span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="n">col</span><span class="o">=</span><span class="w"> </span><span class="s">&quot;blue&quot;</span><span class="p">,</span><span class="n">label</span><span class="o">=</span><span class="nf">paste0</span><span class="p">(</span><span class="s">&quot;n=&quot;</span><span class="p">,</span><span class="nf">length</span><span class="p">(</span><span class="n">downreg</span><span class="p">)))</span><span class="o">+</span>
<span class="w">  </span><span class="nf">annotate</span><span class="p">(</span><span class="s">&quot;text&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">5.5</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="o">=</span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="n">col</span><span class="o">=</span><span class="w"> </span><span class="s">&quot;red&quot;</span><span class="p">,</span><span class="n">label</span><span class="o">=</span><span class="nf">paste0</span><span class="p">(</span><span class="s">&quot;n=&quot;</span><span class="p">,</span><span class="nf">length</span><span class="p">(</span><span class="n">upreg</span><span class="p">)))</span>
</pre></div>
</div>
</li>
</ol>
</section>
<section id="clusterprofiler">
<h2>3. clusterProfiler<a class="headerlink" href="#clusterprofiler" title="Permalink to this heading">#</a></h2>
<p>Used for Gene Ontology, KEGG, Reactome and WikiPathways GSEA or ORA
analysis. Very useful feature for simplifying term results (removing
similar/redundant terms, easing up the interpretation). Used with
‘enrichplot’ to visualize results.</p>
<p><strong>Important note:</strong> This package likes to not cooperate, esp. if it
comes to plots and gseKEGG functions. To resolve the problem consider
these:</p>
<ul class="simple">
<li><p>make sure you have the latest version of BiocManager;</p></li>
<li><p>update clusterProfiler, enrichplot, org db, dependancies etc;</p></li>
<li><p>If still not working, use
“remotes::install_github(”YuLab-SMU/clusterProfiler”))” to download
dev version - last time worked;</p></li>
<li><p>still nothing - download KEGG db and use internal db.</p></li>
</ul>
<ol class="arabic">
<li><p>Load libraries</p>
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">library</span><span class="p">(</span><span class="n">clusterProfiler</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">org.Hs.eg.db</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">enrichplot</span><span class="p">)</span>
</pre></div>
</div>
</li>
<li><p>Prepare geneList which is a named list of genes/proteins with their
log2 fold changes/pvalues from DE like limma’s topTable.</p>
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">geneList</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">DEP</span><span class="p">[,</span><span class="m">2</span><span class="p">]</span><span class="w"> </span><span class="c1">#fc</span>
<span class="nf">names</span><span class="p">(</span><span class="n">geneList</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">as.character</span><span class="p">(</span><span class="n">DEP</span><span class="p">[,</span><span class="m">1</span><span class="p">])</span><span class="w"> </span><span class="c1">#gene symbols</span>
<span class="n">geneList</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">sort</span><span class="p">(</span><span class="n">geneList</span><span class="p">,</span><span class="w"> </span><span class="n">decreasing</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span><span class="w"> </span><span class="c1">#sort ranks</span>
</pre></div>
</div>
</li>
<li><p>Some functions, like ‘gseKEGG’ don’t accept symbols as input, in
this case it is needed to convert symbols into eg. ENTREZID.</p>
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">entrezid</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">bitr</span><span class="p">(</span><span class="nf">as.character</span><span class="p">(</span><span class="n">DEP</span><span class="p">[,</span><span class="m">1</span><span class="p">]),</span><span class="w"> </span>
<span class="w">                 </span><span class="n">fromType</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;SYMBOL&quot;</span><span class="p">,</span><span class="w"> </span>
<span class="w">                 </span><span class="n">toType</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;ENTREZID&quot;</span><span class="p">,</span><span class="w"> </span>
<span class="w">                 </span><span class="n">OrgDb</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">org.Hs.eg.db</span><span class="p">)</span>

<span class="n">geneList2</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">DEP</span><span class="p">,</span><span class="w"> </span><span class="n">ID</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">entrezid</span><span class="o">$</span><span class="n">SYMBOL</span><span class="p">)[,</span><span class="m">2</span><span class="p">]</span><span class="w"> </span><span class="c1">#fc - filter out unmapped</span>
<span class="nf">names</span><span class="p">(</span><span class="n">geneList2</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">as.character</span><span class="p">(</span><span class="n">entrezid</span><span class="o">$</span><span class="n">ENTREZID</span><span class="p">)</span><span class="w"> </span><span class="c1">#gene symbols</span>
<span class="n">geneList2</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">sort</span><span class="p">(</span><span class="n">geneList2</span><span class="p">,</span><span class="w"> </span><span class="n">decreasing</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span><span class="w"> </span><span class="c1">#sort ranks</span>
</pre></div>
</div>
</li>
<li><p>Sometimes you want to use external tools like WebGESTALT. Write down
the geneList into tab-seperated file:</p>
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">write.table</span><span class="p">(</span><span class="n">geneList</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;genelist.txt&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;\t&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">quote</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span>
</pre></div>
</div>
<h3 class="rubric" id="go-gsea">3.1 GO GSEA</h3>
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">gsea</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">gseGO</span><span class="p">(</span><span class="n">geneList</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">geneList</span><span class="p">,</span>
<span class="w">              </span><span class="n">OrgDb</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">org.Hs.eg.db</span><span class="p">,</span>
<span class="w">              </span><span class="n">ont</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;BP&quot;</span><span class="p">,</span><span class="w"> </span><span class="c1"># or CC, MF</span>
<span class="w">              </span><span class="n">minGSSize</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">50</span><span class="p">,</span><span class="w"> </span><span class="c1"># min nr of genes mapped to term</span>
<span class="w">              </span><span class="n">maxGSSize</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">100</span><span class="p">,</span><span class="w"> </span><span class="c1"># max nr of genes mapped to term</span>
<span class="w">              </span><span class="n">pvalueCutoff</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.05</span><span class="p">,</span>
<span class="w">              </span><span class="n">verbose</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span>
<span class="w">              </span><span class="n">keyType</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;SYMBOL&quot;</span><span class="p">)</span>
</pre></div>
</div>
</li>
</ol>
<ul>
<li><p>To view results from GSEA:</p>
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">gsea_results</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">gsea</span><span class="o">@</span><span class="n">result</span>
<span class="n">upreg</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">gsea_results</span><span class="p">,</span><span class="w"> </span><span class="n">NES</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="m">0</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="n">p.adjust</span><span class="w"> </span><span class="o">&lt;</span><span class="w"> </span><span class="m">0.05</span><span class="p">)</span>
<span class="n">downreg</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">gsea_results</span><span class="p">,</span><span class="w"> </span><span class="n">NES</span><span class="w"> </span><span class="o">&lt;</span><span class="w"> </span><span class="m">0</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="n">p.adjust</span><span class="w"> </span><span class="o">&lt;</span><span class="w"> </span><span class="m">0.05</span><span class="p">)</span>
<span class="nf">head</span><span class="p">(</span><span class="n">gsea_results</span><span class="p">)</span>
</pre></div>
</div>
</li>
<li><p>Sometimes it is possible to merge similar terms through ‘simplify()’:</p>
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">simp</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">clusterProfiler</span><span class="o">::</span><span class="nf">simplify</span><span class="p">(</span><span class="n">gsea</span><span class="p">)</span>
<span class="n">simp_results</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">simp</span><span class="o">@</span><span class="n">result</span>
</pre></div>
</div>
<h3 class="rubric" id="kegg-gsea">3.2 KEGG GSEA</h3>
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">kegg</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">clusterProfiler</span><span class="o">::</span><span class="nf">gseKEGG</span><span class="p">(</span><span class="n">geneList</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">geneList2</span><span class="p">,</span>
<span class="w">                </span><span class="n">organism</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;hsa&quot;</span><span class="p">,</span><span class="w"> </span>
<span class="w">                </span><span class="n">keyType</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;ncbi-geneid&quot;</span><span class="p">,</span><span class="w"> </span><span class="c1"># if ENTREZID in geneList</span>
<span class="w">                </span><span class="n">minGSSize</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">20</span><span class="p">,</span>
<span class="w">                </span><span class="n">maxGSSize</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">500</span><span class="p">,</span>
<span class="w">                </span><span class="n">pvalueCutoff</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.05</span><span class="p">,</span>
<span class="w">                </span><span class="n">verbose</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span>

<span class="n">kegg_result</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">kegg</span><span class="o">@</span><span class="n">result</span>
<span class="nf">head</span><span class="p">(</span><span class="n">kegg_result</span><span class="p">)</span>
<span class="c1">#this function converts entrezid back to readable symbols</span>
<span class="n">kegg</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">setReadable</span><span class="p">(</span><span class="n">kegg</span><span class="p">)</span>
</pre></div>
</div>
<h3 class="rubric" id="visualisation-of-gsea-results">3.3 Visualisation of GSEA results</h3>
<p>You can choose from barplots, dotplots, ridgeplots, umaps, cnetplots
etc. Check clustProfiler and enrichplot vignette for possibilities.</p>
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">ridgeplot</span><span class="p">(</span><span class="n">kegg</span><span class="p">)</span><span class="w"> </span><span class="o">+</span>
<span class="w">  </span><span class="nf">geom_vline</span><span class="p">(</span><span class="n">xintercept</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="n">lty</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">2</span><span class="p">)</span>
</pre></div>
</div>
<p>In case you’d like to make manual barplot, which is nicely ordered by
p-value:</p>
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">library</span><span class="p">(</span><span class="n">ggplot2</span><span class="p">)</span>
<span class="nf">theme_set</span><span class="p">(</span><span class="nf">theme_minimal</span><span class="p">())</span>

<span class="n">upreg</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">  </span><span class="nf">arrange</span><span class="p">(</span><span class="nf">desc</span><span class="p">(</span><span class="n">p.adjust</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">mutate</span><span class="p">(</span><span class="n">Description</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">factor</span><span class="p">(</span><span class="n">Description</span><span class="p">,</span><span class="w"> </span><span class="n">levels</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Description</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">ggplot</span><span class="p">()</span><span class="o">+</span>
<span class="w">    </span><span class="nf">geom_col</span><span class="p">(</span><span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">setSize</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Description</span><span class="p">,</span><span class="w"> </span><span class="n">fill</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="o">-</span><span class="nf">log10</span><span class="p">(</span><span class="n">p.adjust</span><span class="p">)))</span><span class="o">+</span>
<span class="w">    </span><span class="nf">scale_fill_gradient</span><span class="p">(</span><span class="n">low</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;blue&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">high</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;red&quot;</span><span class="p">)</span>
</pre></div>
</div>
<h3 class="rubric" id="functions-for-multiple-groups">3.4 Functions for multiple groups</h3>
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># Create a named list with coef numbers for topTable and its name of comparison</span>
<span class="n">coefs</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">seq.int</span><span class="p">(</span><span class="m">1</span><span class="p">,</span><span class="m">3</span><span class="p">,</span><span class="m">1</span><span class="p">)</span>
<span class="nf">names</span><span class="p">(</span><span class="n">coefs</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;ctrlVtrt1&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;ctrlVtrt2&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;trt1Vtrt2&quot;</span><span class="p">)</span>
</pre></div>
</div>
<ul>
<li><p>GO GSEA</p>
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">compute_gsea</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="kr">function</span><span class="p">(</span><span class="n">fit</span><span class="p">,</span><span class="w"> </span><span class="n">GO</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;BP|CC|MF&quot;</span><span class="p">),</span><span class="w"> </span><span class="n">coefs</span><span class="p">,</span><span class="w"> </span><span class="n">simplify</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">  </span><span class="c1">## Case 1. Simplified table results + figures---------</span>
<span class="w">  </span><span class="kr">if</span><span class="w"> </span><span class="p">(</span><span class="n">simplify</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">    </span><span class="n">path_figs</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="s">&quot;figures/gsego_results/gsego_&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">GO</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;_results/unfiltered/&quot;</span><span class="p">)</span>
<span class="w">    </span><span class="n">path_data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="s">&quot;data/gsego_results/gsego_&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">GO</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;_results/unfiltered/simplified/&quot;</span><span class="p">)</span>

<span class="w">    </span><span class="kr">if</span><span class="w"> </span><span class="p">(</span><span class="nf">dir.exists</span><span class="p">(</span><span class="n">path_figs</span><span class="p">)</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">      </span><span class="nf">dir.create</span><span class="p">(</span><span class="n">path_figs</span><span class="p">,</span><span class="n">showWarnings</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="n">recursive</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span>
<span class="w">    </span><span class="p">}</span>

<span class="w">    </span><span class="kr">if</span><span class="w"> </span><span class="p">(</span><span class="nf">dir.exists</span><span class="p">(</span><span class="n">path_data</span><span class="p">)</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">      </span><span class="nf">dir.create</span><span class="p">(</span><span class="n">path_data</span><span class="p">,</span><span class="n">showWarnings</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="n">recursive</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span>
<span class="w">    </span><span class="p">}</span>

<span class="w">    </span><span class="kr">for</span><span class="w"> </span><span class="p">(</span><span class="n">i</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="n">coefs</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">      </span><span class="n">filename_data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="s">&quot;gsego_&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">GO</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;_&quot;</span><span class="p">,</span><span class="w"> </span><span class="nf">names</span><span class="p">(</span><span class="n">coefs</span><span class="p">[</span><span class="n">i</span><span class="p">]),</span><span class="w"> </span><span class="s">&quot;.tsv&quot;</span><span class="p">)</span>
<span class="w">      </span><span class="n">filename_fig</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="s">&quot;gsego_&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">GO</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;_top30_&quot;</span><span class="p">,</span><span class="w"> </span><span class="nf">names</span><span class="p">(</span><span class="n">coefs</span><span class="p">[</span><span class="n">i</span><span class="p">]),</span><span class="w"> </span><span class="s">&quot;.png&quot;</span><span class="p">)</span>

<span class="w">      </span><span class="n">deps</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">topTable</span><span class="p">(</span><span class="n">fit_bayes</span><span class="p">,</span><span class="w"> </span><span class="n">coef</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">i</span><span class="p">,</span><span class="w"> </span><span class="n">adjust</span><span class="o">=</span><span class="s">&quot;BH&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">genelist</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">rownames</span><span class="p">(</span><span class="n">mat</span><span class="p">),</span><span class="w"> </span><span class="n">resort.by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;logFC&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">number</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">nrow</span><span class="p">(</span><span class="n">qnorm</span><span class="p">))</span>

<span class="w">      </span><span class="n">geneList</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">deps</span><span class="p">[,</span><span class="m">2</span><span class="p">]</span><span class="w"> </span><span class="c1">#fc</span>
<span class="w">      </span><span class="nf">names</span><span class="p">(</span><span class="n">geneList</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">as.character</span><span class="p">(</span><span class="n">deps</span><span class="p">[,</span><span class="m">1</span><span class="p">])</span><span class="w"> </span><span class="c1">#gene symbols</span>
<span class="w">      </span><span class="n">geneList</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">sort</span><span class="p">(</span><span class="n">geneList</span><span class="p">,</span><span class="w"> </span><span class="n">decreasing</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span><span class="w"> </span><span class="c1">#sort ranks</span>

<span class="w">      </span><span class="c1"># define keytype for genes</span>
<span class="w">      </span><span class="n">keyType</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;SYMBOL&quot;</span>

<span class="w">      </span><span class="c1"># GO GSEA </span>
<span class="w">      </span><span class="c1"># min/max GSSize means how many genes per term should be used</span>
<span class="w">      </span><span class="n">gsea</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">gseGO</span><span class="p">(</span><span class="n">geneList</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">geneList</span><span class="p">,</span>
<span class="w">                    </span><span class="n">OrgDb</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">org.Hs.eg.db</span><span class="p">,</span>
<span class="w">                    </span><span class="n">ont</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">GO</span><span class="p">,</span>
<span class="w">                    </span><span class="n">minGSSize</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">15</span><span class="p">,</span>
<span class="w">                    </span><span class="n">maxGSSize</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">100</span><span class="p">,</span>
<span class="w">                    </span><span class="n">pvalueCutoff</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.05</span><span class="p">,</span>
<span class="w">                    </span><span class="n">verbose</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span>
<span class="w">                    </span><span class="n">keyType</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">keyType</span><span class="p">)</span>


<span class="w">      </span><span class="c1"># remove redundant GO terms via GOSemSim methods</span>
<span class="w">      </span><span class="n">simp</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">clusterProfiler</span><span class="o">::</span><span class="nf">simplify</span><span class="p">(</span><span class="n">gsea</span><span class="p">)</span>
<span class="w">      </span><span class="n">simp_results</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">simp</span><span class="o">@</span><span class="n">result</span>
<span class="w">      </span><span class="nf">write.table</span><span class="p">(</span><span class="n">file</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="n">path_data</span><span class="p">,</span><span class="w"> </span><span class="n">filename_data</span><span class="p">),</span><span class="w"> </span>
<span class="w">                  </span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">simp_results</span><span class="p">,</span><span class="w"> </span>
<span class="w">                  </span><span class="n">quote</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span><span class="w"> </span>
<span class="w">                  </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;\t&quot;</span><span class="p">,</span>
<span class="w">                  </span><span class="n">row.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span>

<span class="w">      </span><span class="c1"># Visualize</span>
<span class="w">      </span><span class="nf">ridgeplot</span><span class="p">(</span><span class="n">simp</span><span class="p">,</span><span class="w"> </span><span class="n">label_format</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">60</span><span class="p">)</span><span class="o">+</span>
<span class="w">        </span><span class="nf">geom_vline</span><span class="p">(</span><span class="n">xintercept</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="n">lty</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">2</span><span class="p">)</span><span class="o">+</span>
<span class="w">        </span><span class="nf">labs</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;log2FC&quot;</span><span class="p">)</span>

<span class="w">      </span><span class="nf">ggsave</span><span class="p">(</span><span class="n">filename</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">filename_fig</span><span class="p">,</span>
<span class="w">             </span><span class="n">device</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;png&quot;</span><span class="p">,</span>
<span class="w">             </span><span class="n">path</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">path_figs</span><span class="p">,</span>
<span class="w">             </span><span class="n">width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">800</span><span class="p">,</span>
<span class="w">             </span><span class="n">height</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">780</span><span class="p">,</span>
<span class="w">             </span><span class="n">units</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;px&quot;</span><span class="p">,</span>
<span class="w">             </span><span class="n">dpi</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">100</span><span class="p">)</span>
<span class="w">    </span><span class="p">}</span>
<span class="w">  </span><span class="p">}</span><span class="w"> </span><span class="kr">else</span><span class="w"> </span><span class="kr">if</span><span class="w"> </span><span class="p">(</span><span class="n">simplify</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">    </span><span class="c1">## Case 2. NOT simplified table results ONLY---------</span>
<span class="w">      </span><span class="n">path_data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="s">&quot;data/gsego_results/gsego_&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">GO</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;_results/unfiltered/notsimplified/&quot;</span><span class="p">)</span>

<span class="w">      </span><span class="kr">if</span><span class="w"> </span><span class="p">(</span><span class="nf">dir.exists</span><span class="p">(</span><span class="n">path_data</span><span class="p">)</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">        </span><span class="nf">dir.create</span><span class="p">(</span><span class="n">path_data</span><span class="p">,</span><span class="n">showWarnings</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="n">recursive</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span>
<span class="w">      </span><span class="p">}</span>

<span class="w">      </span><span class="kr">for</span><span class="w"> </span><span class="p">(</span><span class="n">i</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="n">coefs</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">        </span><span class="n">filename_data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="s">&quot;gsego_&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">GO</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;_&quot;</span><span class="p">,</span><span class="w"> </span><span class="nf">names</span><span class="p">(</span><span class="n">coefs</span><span class="p">[</span><span class="n">i</span><span class="p">]),</span><span class="w"> </span><span class="s">&quot;.tsv&quot;</span><span class="p">)</span>

<span class="w">        </span><span class="n">deps</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">topTable</span><span class="p">(</span><span class="n">fit_bayes</span><span class="p">,</span><span class="w"> </span><span class="n">coef</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">i</span><span class="p">,</span><span class="w"> </span><span class="n">adjust</span><span class="o">=</span><span class="s">&quot;BH&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">genelist</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">rownames</span><span class="p">(</span><span class="n">mat</span><span class="p">),</span><span class="w"> </span><span class="n">resort.by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;logFC&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">number</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">nrow</span><span class="p">(</span><span class="n">qnorm</span><span class="p">))</span>

<span class="w">        </span><span class="n">geneList</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">deps</span><span class="p">[,</span><span class="m">2</span><span class="p">]</span><span class="w"> </span><span class="c1">#fc</span>
<span class="w">        </span><span class="nf">names</span><span class="p">(</span><span class="n">geneList</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">as.character</span><span class="p">(</span><span class="n">deps</span><span class="p">[,</span><span class="m">1</span><span class="p">])</span><span class="w"> </span><span class="c1">#gene symbols</span>
<span class="w">        </span><span class="n">geneList</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">sort</span><span class="p">(</span><span class="n">geneList</span><span class="p">,</span><span class="w"> </span><span class="n">decreasing</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span><span class="w"> </span><span class="c1">#sort ranks</span>

<span class="w">        </span><span class="c1"># define keytype for genes</span>
<span class="w">        </span><span class="n">keyType</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;SYMBOL&quot;</span>

<span class="w">        </span><span class="c1"># GO GSEA </span>
<span class="w">        </span><span class="c1"># min/max GSSize means how many genes per term should be used</span>
<span class="w">        </span><span class="n">gsea</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">gseGO</span><span class="p">(</span><span class="n">geneList</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">geneList</span><span class="p">,</span>
<span class="w">                      </span><span class="n">OrgDb</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">org.Hs.eg.db</span><span class="p">,</span>
<span class="w">                      </span><span class="n">ont</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">GO</span><span class="p">,</span>
<span class="w">                      </span><span class="n">minGSSize</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">15</span><span class="p">,</span>
<span class="w">                      </span><span class="n">maxGSSize</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">100</span><span class="p">,</span>
<span class="w">                      </span><span class="n">pvalueCutoff</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.05</span><span class="p">,</span>
<span class="w">                      </span><span class="n">verbose</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span>
<span class="w">                      </span><span class="n">keyType</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">keyType</span><span class="p">)</span>


<span class="w">        </span><span class="n">gsea_results</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">gsea</span><span class="o">@</span><span class="n">result</span>
<span class="w">        </span><span class="nf">write.table</span><span class="p">(</span><span class="n">file</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="n">path_data</span><span class="p">,</span><span class="w"> </span><span class="n">filename_data</span><span class="p">),</span><span class="w"> </span>
<span class="w">                    </span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">gsea_results</span><span class="p">,</span><span class="w"> </span>
<span class="w">                    </span><span class="n">quote</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span><span class="w"> </span>
<span class="w">                    </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;\t&quot;</span><span class="p">,</span>
<span class="w">                    </span><span class="n">row.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span>
<span class="w">    </span><span class="p">}</span>
<span class="w">  </span><span class="p">}</span>
<span class="p">}</span>
</pre></div>
</div>
</li>
<li><p>KEGG GSEA</p>
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">compute_keggsea</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="kr">function</span><span class="p">(</span><span class="n">fit</span><span class="p">,</span><span class="w"> </span><span class="n">coefs</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">  </span><span class="n">path_figs</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="s">&quot;figures/gsekegg_results/unfiltered/&quot;</span><span class="p">)</span>
<span class="w">  </span><span class="n">path_data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="s">&quot;data/gsekegg_results/unfiltered/&quot;</span><span class="p">)</span>
<span class="w">  </span><span class="kr">if</span><span class="w"> </span><span class="p">(</span><span class="nf">dir.exists</span><span class="p">(</span><span class="n">path_figs</span><span class="p">)</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">    </span><span class="nf">dir.create</span><span class="p">(</span><span class="n">path_figs</span><span class="p">,</span><span class="n">showWarnings</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="n">recursive</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span>
<span class="w">  </span><span class="p">}</span>

<span class="w">  </span><span class="kr">if</span><span class="w"> </span><span class="p">(</span><span class="nf">dir.exists</span><span class="p">(</span><span class="n">path_data</span><span class="p">)</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">    </span><span class="nf">dir.create</span><span class="p">(</span><span class="n">path_data</span><span class="p">,</span><span class="n">showWarnings</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="n">recursive</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span>
<span class="w">  </span><span class="p">}</span>

<span class="w">  </span><span class="kr">for</span><span class="w"> </span><span class="p">(</span><span class="n">i</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="n">coefs</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">    </span><span class="n">filename_data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="s">&quot;gsekegg_&quot;</span><span class="p">,</span><span class="w"> </span><span class="nf">names</span><span class="p">(</span><span class="n">coefs</span><span class="p">[</span><span class="n">i</span><span class="p">]),</span><span class="w"> </span><span class="s">&quot;.tsv&quot;</span><span class="p">)</span>
<span class="w">    </span><span class="n">filename_fig</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="s">&quot;gsekegg_&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;top30_&quot;</span><span class="p">,</span><span class="w"> </span><span class="nf">names</span><span class="p">(</span><span class="n">coefs</span><span class="p">[</span><span class="n">i</span><span class="p">]),</span><span class="w"> </span><span class="s">&quot;.png&quot;</span><span class="p">)</span>

<span class="w">    </span><span class="n">deps</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">topTable</span><span class="p">(</span><span class="n">fit_bayes</span><span class="p">,</span><span class="w"> </span><span class="n">coef</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">i</span><span class="p">,</span><span class="w"> </span><span class="n">adjust</span><span class="o">=</span><span class="s">&quot;BH&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">genelist</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">rownames</span><span class="p">(</span><span class="n">mat</span><span class="p">),</span><span class="w"> </span><span class="n">resort.by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;logFC&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">number</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">nrow</span><span class="p">(</span><span class="n">qnorm</span><span class="p">))</span>

<span class="w">    </span><span class="n">entrezid</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">bitr</span><span class="p">(</span><span class="nf">as.character</span><span class="p">(</span><span class="n">deps</span><span class="p">[,</span><span class="m">1</span><span class="p">]),</span><span class="w"> </span><span class="n">fromType</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;SYMBOL&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">toType</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;ENTREZID&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">OrgDb</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">org.Hs.eg.db</span><span class="p">)</span>
<span class="w">    </span><span class="n">geneList</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">deps</span><span class="p">,</span><span class="w"> </span><span class="n">ID</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">entrezid</span><span class="o">$</span><span class="n">SYMBOL</span><span class="p">)[,</span><span class="m">2</span><span class="p">]</span><span class="w"> </span><span class="c1">#fc</span>
<span class="w">    </span><span class="nf">names</span><span class="p">(</span><span class="n">geneList</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">as.character</span><span class="p">(</span><span class="n">entrezid</span><span class="o">$</span><span class="n">ENTREZID</span><span class="p">)</span><span class="w"> </span><span class="c1">#gene symbols</span>
<span class="w">    </span><span class="n">geneList</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">sort</span><span class="p">(</span><span class="n">geneList</span><span class="p">,</span><span class="w"> </span><span class="n">decreasing</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span><span class="w"> </span><span class="c1">#sort ranks</span>

<span class="w">    </span><span class="n">kegg</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">gseKEGG</span><span class="p">(</span><span class="n">geneList</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">geneList</span><span class="p">,</span><span class="w">                </span>
<span class="w">                    </span><span class="n">organism</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;hsa&quot;</span><span class="p">,</span><span class="w"> </span>
<span class="w">                    </span><span class="n">keyType</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;ncbi-geneid&quot;</span><span class="p">,</span>
<span class="w">                    </span><span class="n">minGSSize</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">20</span><span class="p">,</span>
<span class="w">                    </span><span class="n">maxGSSize</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">100</span><span class="p">,</span>
<span class="w">                    </span><span class="n">pvalueCutoff</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.05</span><span class="p">,</span>
<span class="w">                    </span><span class="n">verbose</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span>

<span class="w">    </span><span class="n">kegg</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">setReadable</span><span class="p">(</span><span class="n">kegg</span><span class="p">,</span><span class="w"> </span><span class="n">OrgDb</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">org.Hs.eg.db</span><span class="p">,</span><span class="w"> </span><span class="n">keyType</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;ENTREZID&quot;</span><span class="p">)</span>

<span class="w">    </span><span class="n">kegg_results</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">kegg</span><span class="o">@</span><span class="n">result</span>
<span class="w">    </span><span class="nf">write.table</span><span class="p">(</span><span class="n">file</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="n">path_data</span><span class="p">,</span><span class="w"> </span><span class="n">filename_data</span><span class="p">),</span><span class="w"> </span>
<span class="w">                </span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">kegg_results</span><span class="p">,</span><span class="w"> </span>
<span class="w">                </span><span class="n">quote</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span><span class="w"> </span>
<span class="w">                </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;\t&quot;</span><span class="p">,</span>
<span class="w">                </span><span class="n">row.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span>

<span class="w">    </span><span class="c1"># Visualize</span>
<span class="w">    </span><span class="nf">ridgeplot</span><span class="p">(</span><span class="n">kegg</span><span class="p">,</span><span class="w"> </span><span class="n">label_format</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">60</span><span class="p">)</span><span class="o">+</span>
<span class="w">      </span><span class="nf">geom_vline</span><span class="p">(</span><span class="n">xintercept</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="n">lty</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">2</span><span class="p">)</span><span class="o">+</span>
<span class="w">      </span><span class="nf">labs</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;log2FC&quot;</span><span class="p">)</span>

<span class="w">    </span><span class="nf">ggsave</span><span class="p">(</span><span class="n">filename</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">filename_fig</span><span class="p">,</span>
<span class="w">           </span><span class="n">device</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;png&quot;</span><span class="p">,</span>
<span class="w">           </span><span class="n">path</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">path_figs</span><span class="p">,</span>
<span class="w">           </span><span class="n">width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">800</span><span class="p">,</span>
<span class="w">           </span><span class="n">height</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">780</span><span class="p">,</span>
<span class="w">           </span><span class="n">units</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;px&quot;</span><span class="p">,</span>
<span class="w">           </span><span class="n">dpi</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">100</span><span class="p">)</span>
<span class="w">  </span><span class="p">}</span>
<span class="p">}</span>
</pre></div>
</div>
</li>
<li><p>GO ORA</p>
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">enrich_go</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="kr">function</span><span class="p">(</span><span class="n">fit</span><span class="p">,</span><span class="w"> </span><span class="n">coefs</span><span class="p">,</span><span class="w"> </span><span class="n">GO</span><span class="o">=</span><span class="s">&quot;BP|CC|MF&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">simplify</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">  </span><span class="c1"># simplify=TRUE -------------------------</span>
<span class="w">  </span><span class="kr">if</span><span class="w"> </span><span class="p">(</span><span class="n">simplify</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">    </span><span class="n">path_figs</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="s">&quot;figures/enrichgo_results/enrich_&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">GO</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;_results/unfiltered/&quot;</span><span class="p">)</span>
<span class="w">    </span><span class="n">path_data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="s">&quot;data/enrichgo_results/enrich_&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">GO</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;_results/unfiltered/simplified/&quot;</span><span class="p">)</span>

<span class="w">    </span><span class="kr">if</span><span class="w"> </span><span class="p">(</span><span class="nf">dir.exists</span><span class="p">(</span><span class="n">path_figs</span><span class="p">)</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">      </span><span class="nf">dir.create</span><span class="p">(</span><span class="n">path_figs</span><span class="p">,</span><span class="n">showWarnings</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="n">recursive</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span>
<span class="w">    </span><span class="p">}</span>

<span class="w">    </span><span class="kr">if</span><span class="w"> </span><span class="p">(</span><span class="nf">dir.exists</span><span class="p">(</span><span class="n">path_data</span><span class="p">)</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">      </span><span class="nf">dir.create</span><span class="p">(</span><span class="n">path_data</span><span class="p">,</span><span class="n">showWarnings</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="n">recursive</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span>
<span class="w">    </span><span class="p">}</span>

<span class="w">    </span><span class="kr">for</span><span class="w"> </span><span class="p">(</span><span class="n">i</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="n">coefs</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">      </span><span class="n">filename_data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="s">&quot;enrich_&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">GO</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;_sim_&quot;</span><span class="p">,</span><span class="w"> </span><span class="nf">names</span><span class="p">(</span><span class="n">coefs</span><span class="p">[</span><span class="n">i</span><span class="p">]))</span>
<span class="w">      </span><span class="n">filename_fig</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="s">&quot;enrich_&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">GO</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;_top30_sim_&quot;</span><span class="p">,</span><span class="w"> </span><span class="nf">names</span><span class="p">(</span><span class="n">coefs</span><span class="p">[</span><span class="n">i</span><span class="p">]))</span>

<span class="w">      </span><span class="n">deps</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">topTable</span><span class="p">(</span><span class="n">fit_bayes</span><span class="p">,</span><span class="w"> </span><span class="n">coef</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">i</span><span class="p">,</span><span class="w"> </span><span class="n">adjust</span><span class="o">=</span><span class="s">&quot;BH&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">genelist</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">rownames</span><span class="p">(</span><span class="n">mat</span><span class="p">),</span><span class="w"> </span><span class="n">resort.by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;logFC&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">number</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">nrow</span><span class="p">(</span><span class="n">qnorm</span><span class="p">))</span>

<span class="w">      </span><span class="n">geneList</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">deps</span><span class="p">[,</span><span class="m">2</span><span class="p">]</span><span class="w"> </span><span class="c1">#fc</span>
<span class="w">      </span><span class="nf">names</span><span class="p">(</span><span class="n">geneList</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">as.character</span><span class="p">(</span><span class="n">deps</span><span class="p">[,</span><span class="m">1</span><span class="p">])</span><span class="w"> </span><span class="c1">#gene symbols</span>
<span class="w">      </span><span class="n">geneList</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">sort</span><span class="p">(</span><span class="n">geneList</span><span class="p">,</span><span class="w"> </span><span class="n">decreasing</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span><span class="w"> </span><span class="c1">#sort ranks</span>

<span class="w">      </span><span class="c1"># define keytype for genes</span>
<span class="w">      </span><span class="n">keyType</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;SYMBOL&quot;</span>

<span class="w">      </span><span class="c1">#extract ids</span>
<span class="w">      </span><span class="n">upreg</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">deps</span><span class="o">$</span><span class="n">ID</span><span class="p">[</span><span class="n">deps</span><span class="o">$</span><span class="n">logFC</span><span class="w"> </span><span class="o">&gt;=</span><span class="w"> </span><span class="m">1.3</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="n">deps</span><span class="o">$</span><span class="n">adj.P.Val</span><span class="w"> </span><span class="o">&lt;=</span><span class="w"> </span><span class="m">0.05</span><span class="p">]</span>
<span class="w">      </span><span class="n">downreg</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">deps</span><span class="o">$</span><span class="n">ID</span><span class="p">[</span><span class="n">deps</span><span class="o">$</span><span class="n">logFC</span><span class="w"> </span><span class="o">&lt;=</span><span class="w"> </span><span class="m">-1.3</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="n">deps</span><span class="o">$</span><span class="n">adj.P.Val</span><span class="w"> </span><span class="o">&lt;=</span><span class="w"> </span><span class="m">0.05</span><span class="p">]</span>
<span class="w">      </span><span class="nf">message</span><span class="p">(</span><span class="nf">paste0</span><span class="p">(</span><span class="nf">names</span><span class="p">(</span><span class="n">coefs</span><span class="p">[</span><span class="n">i</span><span class="p">]),</span><span class="s">&quot;: &quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;Nr of upregulated genes:&quot;</span><span class="p">,</span><span class="w"> </span><span class="nf">length</span><span class="p">(</span><span class="n">upreg</span><span class="p">)))</span>
<span class="w">      </span><span class="nf">message</span><span class="p">(</span><span class="nf">paste0</span><span class="p">(</span><span class="nf">names</span><span class="p">(</span><span class="n">coefs</span><span class="p">[</span><span class="n">i</span><span class="p">]),</span><span class="s">&quot;: &quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;Nr of downregulated genes:&quot;</span><span class="p">,</span><span class="w"> </span><span class="nf">length</span><span class="p">(</span><span class="n">downreg</span><span class="p">)))</span>

<span class="w">      </span><span class="c1">#enrich upreg</span>
<span class="w">      </span><span class="n">go_enr</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">enrichGO</span><span class="p">(</span><span class="n">gene</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">upreg</span><span class="p">,</span>
<span class="w">                         </span><span class="n">universe</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">names</span><span class="p">(</span><span class="n">geneList</span><span class="p">),</span>
<span class="w">                         </span><span class="n">OrgDb</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">org.Hs.eg.db</span><span class="p">,</span>
<span class="w">                         </span><span class="n">keyType</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;SYMBOL&quot;</span><span class="p">,</span>
<span class="w">                         </span><span class="n">ont</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">GO</span><span class="p">,</span>
<span class="w">                         </span><span class="n">pvalueCutoff</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.05</span><span class="p">,</span>
<span class="w">                         </span><span class="n">pAdjustMethod</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;BH&quot;</span><span class="p">)</span>
<span class="w">      </span><span class="n">go_enr_results</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">go_enr</span><span class="o">@</span><span class="n">result</span>
<span class="w">      </span><span class="nf">write.table</span><span class="p">(</span><span class="n">file</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="n">path_data</span><span class="p">,</span><span class="w"> </span><span class="n">filename_data</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;_upreg.tsv&quot;</span><span class="p">),</span><span class="w"> </span>
<span class="w">                  </span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">go_enr_results</span><span class="p">,</span><span class="w"> </span>
<span class="w">                  </span><span class="n">quote</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span><span class="w"> </span>
<span class="w">                  </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;\t&quot;</span><span class="p">,</span>
<span class="w">                  </span><span class="n">dec</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;,&quot;</span><span class="p">,</span>
<span class="w">                  </span><span class="n">row.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span>
<span class="w">      </span><span class="c1">#Visualize upreg</span>
<span class="w">      </span><span class="kr">if</span><span class="p">(</span><span class="nf">min</span><span class="p">(</span><span class="n">go_enr_results</span><span class="o">$</span><span class="n">p.adjust</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;=</span><span class="w"> </span><span class="m">0.05</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">        </span><span class="n">go_enr_sim</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">simplify</span><span class="p">(</span><span class="n">go_enr</span><span class="p">)</span>
<span class="w">        </span><span class="nf">barplot</span><span class="p">(</span><span class="n">go_enr_sim</span><span class="p">,</span><span class="w"> </span><span class="n">showCategory</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">30</span><span class="p">,</span><span class="w"> </span><span class="n">label_format</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">60</span><span class="p">)</span>
<span class="w">        </span><span class="nf">ggsave</span><span class="p">(</span><span class="n">filename</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="n">filename_fig</span><span class="p">,</span><span class="s">&quot;_upreg.png&quot;</span><span class="p">),</span>
<span class="w">               </span><span class="n">device</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;png&quot;</span><span class="p">,</span>
<span class="w">               </span><span class="n">path</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">path_figs</span><span class="p">,</span>
<span class="w">               </span><span class="n">width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">800</span><span class="p">,</span>
<span class="w">               </span><span class="n">height</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">780</span><span class="p">,</span>
<span class="w">               </span><span class="n">units</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;px&quot;</span><span class="p">,</span>
<span class="w">               </span><span class="n">dpi</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">100</span><span class="p">)</span>
<span class="w">      </span><span class="p">}</span><span class="w"> </span><span class="kr">else</span><span class="w"> </span><span class="p">{</span>
<span class="w">        </span><span class="nf">message</span><span class="p">(</span><span class="nf">paste0</span><span class="p">(</span><span class="s">&quot;No enrichment for upregulated genes in &quot;</span><span class="p">,</span><span class="w"> </span><span class="nf">names</span><span class="p">(</span><span class="n">coefs</span><span class="p">[</span><span class="n">i</span><span class="p">])))</span>
<span class="w">      </span><span class="p">}</span>
<span class="w">      </span><span class="c1">#enrich downreg</span>
<span class="w">      </span><span class="n">go_enr</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">enrichGO</span><span class="p">(</span><span class="n">gene</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">downreg</span><span class="p">,</span>
<span class="w">                         </span><span class="n">universe</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">names</span><span class="p">(</span><span class="n">geneList</span><span class="p">),</span>
<span class="w">                         </span><span class="n">OrgDb</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">org.Hs.eg.db</span><span class="p">,</span>
<span class="w">                         </span><span class="n">keyType</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;SYMBOL&quot;</span><span class="p">,</span>
<span class="w">                         </span><span class="n">ont</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">GO</span><span class="p">,</span>
<span class="w">                         </span><span class="n">pvalueCutoff</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.05</span><span class="p">,</span>
<span class="w">                         </span><span class="n">pAdjustMethod</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;BH&quot;</span><span class="p">)</span>
<span class="w">      </span><span class="n">go_enr_results</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">go_enr</span><span class="o">@</span><span class="n">result</span>
<span class="w">      </span><span class="nf">write.table</span><span class="p">(</span><span class="n">file</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="n">path_data</span><span class="p">,</span><span class="w"> </span><span class="n">filename_data</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;_downreg.tsv&quot;</span><span class="p">),</span><span class="w"> </span>
<span class="w">                  </span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">go_enr_results</span><span class="p">,</span><span class="w"> </span>
<span class="w">                  </span><span class="n">quote</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span><span class="w"> </span>
<span class="w">                  </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;\t&quot;</span><span class="p">,</span>
<span class="w">                  </span><span class="n">dec</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;,&quot;</span><span class="p">,</span>
<span class="w">                  </span><span class="n">row.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span>
<span class="w">      </span><span class="c1">#Visualize downreg</span>
<span class="w">      </span><span class="kr">if</span><span class="p">(</span><span class="nf">min</span><span class="p">(</span><span class="n">go_enr_results</span><span class="o">$</span><span class="n">p.adjust</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;=</span><span class="w"> </span><span class="m">0.05</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">        </span><span class="n">go_enr_sim</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">simplify</span><span class="p">(</span><span class="n">go_enr</span><span class="p">)</span>
<span class="w">        </span><span class="nf">barplot</span><span class="p">(</span><span class="n">go_enr_sim</span><span class="p">,</span><span class="w"> </span><span class="n">showCategory</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">30</span><span class="p">,</span><span class="w"> </span><span class="n">label_format</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">60</span><span class="p">)</span>
<span class="w">        </span><span class="nf">ggsave</span><span class="p">(</span><span class="n">filename</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="n">filename_fig</span><span class="p">,</span><span class="s">&quot;_downreg.png&quot;</span><span class="p">),</span>
<span class="w">               </span><span class="n">device</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;png&quot;</span><span class="p">,</span>
<span class="w">               </span><span class="n">path</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">path_figs</span><span class="p">,</span>
<span class="w">               </span><span class="n">width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">800</span><span class="p">,</span>
<span class="w">               </span><span class="n">height</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">780</span><span class="p">,</span>
<span class="w">               </span><span class="n">units</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;px&quot;</span><span class="p">,</span>
<span class="w">               </span><span class="n">dpi</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">100</span><span class="p">)</span>
<span class="w">      </span><span class="p">}</span><span class="w"> </span><span class="kr">else</span><span class="w"> </span><span class="p">{</span>
<span class="w">        </span><span class="nf">message</span><span class="p">(</span><span class="nf">paste0</span><span class="p">(</span><span class="s">&quot;No enrichment for downregulated genes in &quot;</span><span class="p">,</span><span class="w"> </span><span class="nf">names</span><span class="p">(</span><span class="n">coefs</span><span class="p">[</span><span class="n">i</span><span class="p">])))</span>
<span class="w">      </span><span class="p">}</span>
<span class="w">      </span><span class="c1"># simplify=FALSE ----------------------------------------------</span>
<span class="w">    </span><span class="p">}</span><span class="w"> </span>
<span class="w">  </span><span class="p">}</span><span class="w"> </span><span class="kr">else</span><span class="w"> </span><span class="kr">if</span><span class="w"> </span><span class="p">(</span><span class="n">simplify</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">    </span><span class="n">path_data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="s">&quot;data/enrichgo_results/enrich_&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">GO</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;_results/unfiltered/not_simplified/&quot;</span><span class="p">)</span>

<span class="w">    </span><span class="kr">if</span><span class="w"> </span><span class="p">(</span><span class="nf">dir.exists</span><span class="p">(</span><span class="n">path_figs</span><span class="p">)</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">      </span><span class="nf">dir.create</span><span class="p">(</span><span class="n">path_figs</span><span class="p">,</span><span class="n">showWarnings</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="n">recursive</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span>
<span class="w">    </span><span class="p">}</span>

<span class="w">    </span><span class="kr">if</span><span class="w"> </span><span class="p">(</span><span class="nf">dir.exists</span><span class="p">(</span><span class="n">path_data</span><span class="p">)</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">      </span><span class="nf">dir.create</span><span class="p">(</span><span class="n">path_data</span><span class="p">,</span><span class="n">showWarnings</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="n">recursive</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span>
<span class="w">    </span><span class="p">}</span>

<span class="w">    </span><span class="kr">for</span><span class="w"> </span><span class="p">(</span><span class="n">i</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="n">coefs</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">      </span><span class="n">filename_data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="s">&quot;enrich_&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">GO</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;_notsim_&quot;</span><span class="p">,</span><span class="w"> </span><span class="nf">names</span><span class="p">(</span><span class="n">coefs</span><span class="p">[</span><span class="n">i</span><span class="p">]))</span>

<span class="w">      </span><span class="n">deps</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">topTable</span><span class="p">(</span><span class="n">fit_bayes</span><span class="p">,</span><span class="w"> </span><span class="n">coef</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">i</span><span class="p">,</span><span class="w"> </span><span class="n">adjust</span><span class="o">=</span><span class="s">&quot;BH&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">genelist</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">rownames</span><span class="p">(</span><span class="n">mat</span><span class="p">),</span><span class="w"> </span><span class="n">resort.by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;logFC&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">number</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">nrow</span><span class="p">(</span><span class="n">qnorm</span><span class="p">))</span>

<span class="w">      </span><span class="n">geneList</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">deps</span><span class="p">[,</span><span class="m">2</span><span class="p">]</span><span class="w"> </span><span class="c1">#fc</span>
<span class="w">      </span><span class="nf">names</span><span class="p">(</span><span class="n">geneList</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">as.character</span><span class="p">(</span><span class="n">deps</span><span class="p">[,</span><span class="m">1</span><span class="p">])</span><span class="w"> </span><span class="c1">#gene symbols</span>
<span class="w">      </span><span class="n">geneList</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">sort</span><span class="p">(</span><span class="n">geneList</span><span class="p">,</span><span class="w"> </span><span class="n">decreasing</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span><span class="w"> </span><span class="c1">#sort ranks</span>

<span class="w">      </span><span class="c1"># define keytype for genes</span>
<span class="w">      </span><span class="n">keyType</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;SYMBOL&quot;</span>

<span class="w">      </span><span class="c1">#extract ids</span>
<span class="w">      </span><span class="n">upreg</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">deps</span><span class="o">$</span><span class="n">ID</span><span class="p">[</span><span class="n">deps</span><span class="o">$</span><span class="n">logFC</span><span class="w"> </span><span class="o">&gt;=</span><span class="w"> </span><span class="m">1.3</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="n">deps</span><span class="o">$</span><span class="n">adj.P.Val</span><span class="w"> </span><span class="o">&lt;=</span><span class="w"> </span><span class="m">0.05</span><span class="p">]</span>
<span class="w">      </span><span class="n">downreg</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">deps</span><span class="o">$</span><span class="n">ID</span><span class="p">[</span><span class="n">deps</span><span class="o">$</span><span class="n">logFC</span><span class="w"> </span><span class="o">&lt;=</span><span class="w"> </span><span class="m">-1.3</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="n">deps</span><span class="o">$</span><span class="n">adj.P.Val</span><span class="w"> </span><span class="o">&lt;=</span><span class="w"> </span><span class="m">0.05</span><span class="p">]</span>
<span class="w">      </span><span class="nf">message</span><span class="p">(</span><span class="nf">paste0</span><span class="p">(</span><span class="nf">names</span><span class="p">(</span><span class="n">coefs</span><span class="p">[</span><span class="n">i</span><span class="p">]),</span><span class="s">&quot;: &quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;Nr of upregulated genes:&quot;</span><span class="p">,</span><span class="w"> </span><span class="nf">length</span><span class="p">(</span><span class="n">upreg</span><span class="p">)))</span>
<span class="w">      </span><span class="nf">message</span><span class="p">(</span><span class="nf">paste0</span><span class="p">(</span><span class="nf">names</span><span class="p">(</span><span class="n">coefs</span><span class="p">[</span><span class="n">i</span><span class="p">]),</span><span class="s">&quot;: &quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;Nr of downregulated genes:&quot;</span><span class="p">,</span><span class="w"> </span><span class="nf">length</span><span class="p">(</span><span class="n">downreg</span><span class="p">)))</span>

<span class="w">      </span><span class="c1">#enrich upreg</span>
<span class="w">      </span><span class="n">go_enr</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">enrichGO</span><span class="p">(</span><span class="n">gene</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">upreg</span><span class="p">,</span>
<span class="w">                         </span><span class="n">universe</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">names</span><span class="p">(</span><span class="n">geneList</span><span class="p">),</span>
<span class="w">                         </span><span class="n">OrgDb</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">org.Hs.eg.db</span><span class="p">,</span>
<span class="w">                         </span><span class="n">keyType</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;SYMBOL&quot;</span><span class="p">,</span>
<span class="w">                         </span><span class="n">ont</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">GO</span><span class="p">,</span>
<span class="w">                         </span><span class="n">pvalueCutoff</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.05</span><span class="p">,</span>
<span class="w">                         </span><span class="n">pAdjustMethod</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;BH&quot;</span><span class="p">)</span>
<span class="w">      </span><span class="n">go_enr_results</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">go_enr</span><span class="o">@</span><span class="n">result</span>
<span class="w">      </span><span class="nf">write.table</span><span class="p">(</span><span class="n">file</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="n">path_data</span><span class="p">,</span><span class="w"> </span><span class="n">filename_data</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;_notsim_upreg.tsv&quot;</span><span class="p">),</span><span class="w"> </span>
<span class="w">                  </span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">go_enr_results</span><span class="p">,</span><span class="w"> </span>
<span class="w">                  </span><span class="n">quote</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span><span class="w"> </span>
<span class="w">                  </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;\t&quot;</span><span class="p">,</span>
<span class="w">                  </span><span class="n">dec</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;,&quot;</span><span class="p">,</span>
<span class="w">                  </span><span class="n">row.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span>

<span class="w">      </span><span class="kr">if</span><span class="p">(</span><span class="nf">min</span><span class="p">(</span><span class="n">go_enr_results</span><span class="o">$</span><span class="n">p.adjust</span><span class="p">)</span><span class="w"> </span><span class="o">&gt;=</span><span class="w"> </span><span class="m">0.05</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">        </span><span class="nf">message</span><span class="p">(</span><span class="nf">paste0</span><span class="p">(</span><span class="s">&quot;No enrichment for upregulated genes in &quot;</span><span class="p">,</span><span class="w"> </span><span class="nf">names</span><span class="p">(</span><span class="n">coefs</span><span class="p">[</span><span class="n">i</span><span class="p">])))</span>
<span class="w">      </span><span class="p">}</span><span class="w"> </span>
<span class="w">      </span><span class="c1">#enrich downreg</span>
<span class="w">      </span><span class="n">go_enr</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">enrichGO</span><span class="p">(</span><span class="n">gene</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">downreg</span><span class="p">,</span>
<span class="w">                         </span><span class="n">universe</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">names</span><span class="p">(</span><span class="n">geneList</span><span class="p">),</span>
<span class="w">                         </span><span class="n">OrgDb</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">org.Hs.eg.db</span><span class="p">,</span>
<span class="w">                         </span><span class="n">keyType</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;SYMBOL&quot;</span><span class="p">,</span>
<span class="w">                         </span><span class="n">ont</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">GO</span><span class="p">,</span>
<span class="w">                         </span><span class="n">pvalueCutoff</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.05</span><span class="p">,</span>
<span class="w">                         </span><span class="n">pAdjustMethod</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;BH&quot;</span><span class="p">)</span>
<span class="w">      </span><span class="n">go_enr_results</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">go_enr</span><span class="o">@</span><span class="n">result</span>
<span class="w">      </span><span class="nf">write.table</span><span class="p">(</span><span class="n">file</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="n">path_data</span><span class="p">,</span><span class="w"> </span><span class="n">filename_data</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;_notsim_downreg.tsv&quot;</span><span class="p">),</span><span class="w"> </span>
<span class="w">                  </span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">go_enr_results</span><span class="p">,</span><span class="w"> </span>
<span class="w">                  </span><span class="n">quote</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span><span class="w"> </span>
<span class="w">                  </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;\t&quot;</span><span class="p">,</span>
<span class="w">                  </span><span class="n">dec</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;,&quot;</span><span class="p">,</span>
<span class="w">                  </span><span class="n">row.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span>
<span class="w">      </span><span class="kr">if</span><span class="p">(</span><span class="nf">min</span><span class="p">(</span><span class="n">go_enr_results</span><span class="o">$</span><span class="n">p.adjust</span><span class="p">)</span><span class="w"> </span><span class="o">&gt;=</span><span class="w"> </span><span class="m">0.05</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">        </span><span class="nf">message</span><span class="p">(</span><span class="nf">paste0</span><span class="p">(</span><span class="s">&quot;No enrichment for downregulated genes in &quot;</span><span class="p">,</span><span class="w"> </span><span class="nf">names</span><span class="p">(</span><span class="n">coefs</span><span class="p">[</span><span class="n">i</span><span class="p">])))</span>
<span class="w">      </span><span class="p">}</span>
<span class="w">    </span><span class="p">}</span>
<span class="w">  </span><span class="p">}</span><span class="w">  </span>
<span class="p">}</span>
</pre></div>
</div>
</li>
<li><p>Volcanoes</p>
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="kr">for</span><span class="w"> </span><span class="p">(</span><span class="n">i</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="n">coefs</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">  </span><span class="kr">if</span><span class="w"> </span><span class="p">(</span><span class="nf">dir.exists</span><span class="p">(</span><span class="s">&quot;figures/volcanoes/&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">    </span><span class="nf">dir.create</span><span class="p">(</span><span class="s">&quot;figures/volcanoes/&quot;</span><span class="p">,</span><span class="n">showWarnings</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="n">recursive</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span>
<span class="w">  </span><span class="p">}</span>
<span class="w">  </span><span class="kr">if</span><span class="w"> </span><span class="p">(</span><span class="nf">dir.exists</span><span class="p">(</span><span class="s">&quot;data/comparison_stats/&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">    </span><span class="nf">dir.create</span><span class="p">(</span><span class="s">&quot;data/comparison_stats/&quot;</span><span class="p">,</span><span class="n">showWarnings</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="n">recursive</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span>
<span class="w">  </span><span class="p">}</span>

<span class="w">  </span><span class="n">deps</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">topTable</span><span class="p">(</span><span class="n">fit_bayes</span><span class="p">,</span><span class="w"> </span><span class="n">coef</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">i</span><span class="p">,</span><span class="w"> </span><span class="n">adjust</span><span class="o">=</span><span class="s">&quot;BH&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">genelist</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">rownames</span><span class="p">(</span><span class="n">mat</span><span class="p">),</span><span class="w"> </span><span class="n">resort.by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;logFC&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">number</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">nrow</span><span class="p">(</span><span class="n">qnorm</span><span class="p">))</span>

<span class="w">  </span><span class="n">volc_data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">deps</span>
<span class="w">  </span><span class="n">volc_data</span><span class="o">$</span><span class="n">Category</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&quot;Not significant&quot;</span>
<span class="w">  </span><span class="n">volc_data</span><span class="o">$</span><span class="n">Category</span><span class="p">[</span><span class="n">volc_data</span><span class="o">$</span><span class="n">logFC</span><span class="w"> </span><span class="o">&lt;=</span><span class="w"> </span><span class="m">-1.3</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="n">volc_data</span><span class="o">$</span><span class="n">adj.P.Val</span><span class="w"> </span><span class="o">&lt;=</span><span class="w"> </span><span class="m">0.05</span><span class="p">]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&quot;Down-regulated&quot;</span>
<span class="w">  </span><span class="n">volc_data</span><span class="o">$</span><span class="n">Category</span><span class="p">[</span><span class="n">volc_data</span><span class="o">$</span><span class="n">logFC</span><span class="w"> </span><span class="o">&gt;=</span><span class="w"> </span><span class="m">1.3</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="n">volc_data</span><span class="o">$</span><span class="n">adj.P.Val</span><span class="w"> </span><span class="o">&lt;=</span><span class="w"> </span><span class="m">0.05</span><span class="p">]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&quot;Up-regulated&quot;</span>
<span class="w">  </span><span class="n">volc_data</span><span class="o">$</span><span class="n">Label</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="kc">NA</span>
<span class="w">  </span><span class="n">volc_data</span><span class="o">$</span><span class="n">Label</span><span class="p">[</span><span class="n">volc_data</span><span class="o">$</span><span class="n">Category</span><span class="w"> </span><span class="o">!=</span><span class="w"> </span><span class="s">&quot;Not significant&quot;</span><span class="p">]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">volc_data</span><span class="o">$</span><span class="n">ID</span><span class="p">[</span><span class="n">volc_data</span><span class="o">$</span><span class="n">Category</span><span class="w"> </span><span class="o">!=</span><span class="w"> </span><span class="s">&quot;Not significant&quot;</span><span class="p">]</span>

<span class="w">  </span><span class="n">upreg</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">volc_data</span><span class="o">$</span><span class="n">ID</span><span class="p">[</span><span class="n">volc_data</span><span class="o">$</span><span class="n">logFC</span><span class="w"> </span><span class="o">&gt;=</span><span class="w"> </span><span class="m">1.3</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="n">volc_data</span><span class="o">$</span><span class="n">adj.P.Val</span><span class="w"> </span><span class="o">&lt;=</span><span class="w"> </span><span class="m">0.05</span><span class="p">]</span>
<span class="w">  </span><span class="n">downreg</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">volc_data</span><span class="o">$</span><span class="n">ID</span><span class="p">[</span><span class="n">volc_data</span><span class="o">$</span><span class="n">logFC</span><span class="w"> </span><span class="o">&lt;=</span><span class="w"> </span><span class="m">-1.3</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="n">volc_data</span><span class="o">$</span><span class="n">adj.P.Val</span><span class="w"> </span><span class="o">&lt;=</span><span class="w"> </span><span class="m">0.05</span><span class="p">]</span>

<span class="w">  </span><span class="nf">write.table</span><span class="p">(</span><span class="n">file</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="s">&quot;data/comparison_stats/&quot;</span><span class="p">,</span><span class="w"> </span><span class="nf">names</span><span class="p">(</span><span class="n">coefs</span><span class="p">[</span><span class="n">i</span><span class="p">]),</span><span class="w"> </span><span class="s">&quot;.tsv&quot;</span><span class="p">),</span><span class="w"> </span>
<span class="w">              </span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">volc_data</span><span class="p">,</span><span class="w"> </span>
<span class="w">              </span><span class="n">quote</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span><span class="w"> </span>
<span class="w">              </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;\t&quot;</span><span class="p">,</span>
<span class="w">              </span><span class="n">row.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span>

<span class="w">  </span><span class="nf">theme_set</span><span class="p">(</span><span class="nf">theme_light</span><span class="p">())</span>
<span class="w">  </span><span class="n">v</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">ggplot</span><span class="p">(</span><span class="n">volc_data</span><span class="p">,</span>
<span class="w">              </span><span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">logFC</span><span class="p">,</span>
<span class="w">                  </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="p">(</span><span class="m">-1</span><span class="o">*</span><span class="nf">log10</span><span class="p">(</span><span class="n">adj.P.Val</span><span class="p">)),</span>
<span class="w">                  </span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Category</span><span class="p">,</span>
<span class="w">                  </span><span class="n">label</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Label</span><span class="p">))</span><span class="o">+</span>
<span class="w">    </span><span class="nf">geom_point</span><span class="p">(</span><span class="n">alpha</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">.</span><span class="m">5</span><span class="p">)</span><span class="o">+</span>
<span class="w">    </span><span class="nf">scale_color_manual</span><span class="p">(</span><span class="n">values</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;Down-regulated&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;blue&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;Not significant&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;gray&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;Up-regulated&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;red&quot;</span><span class="p">))</span><span class="o">+</span>
<span class="w">    </span><span class="nf">labs</span><span class="p">(</span><span class="n">title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">names</span><span class="p">(</span><span class="n">coefs</span><span class="p">[</span><span class="n">i</span><span class="p">]),</span>
<span class="w">         </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;-log10 adj. p-value&quot;</span><span class="p">,</span>
<span class="w">         </span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;log2 fold change&quot;</span><span class="p">)</span><span class="o">+</span>
<span class="w">    </span><span class="nf">geom_hline</span><span class="p">(</span><span class="n">yintercept</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1.3</span><span class="p">,</span><span class="w"> </span><span class="n">lty</span><span class="o">=</span><span class="m">2</span><span class="p">)</span><span class="o">+</span>
<span class="w">    </span><span class="nf">geom_vline</span><span class="p">(</span><span class="n">xintercept</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">-1.3</span><span class="p">,</span><span class="m">1.3</span><span class="p">),</span><span class="w"> </span><span class="n">lty</span><span class="o">=</span><span class="m">2</span><span class="p">)</span><span class="o">+</span>
<span class="w">    </span><span class="nf">scale_x_continuous</span><span class="p">(</span><span class="n">limits</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">-5.5</span><span class="p">,</span><span class="m">5.5</span><span class="p">))</span><span class="o">+</span>
<span class="w">    </span><span class="nf">scale_y_continuous</span><span class="p">(</span><span class="n">limits</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">10.5</span><span class="p">))</span>
<span class="w">  </span><span class="n">v</span><span class="o">+</span>
<span class="w">    </span><span class="nf">theme</span><span class="p">(</span><span class="n">legend.position</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;top&quot;</span><span class="p">,</span>
<span class="w">          </span><span class="n">text</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">element_text</span><span class="p">(</span><span class="n">size</span><span class="o">=</span><span class="m">15</span><span class="p">))</span><span class="o">+</span>
<span class="w">    </span><span class="nf">geom_text_repel</span><span class="p">(</span><span class="n">colour</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;black&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">max.overlaps</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">15</span><span class="p">,</span><span class="w"> </span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">3.5</span><span class="p">)</span><span class="o">+</span>
<span class="w">    </span><span class="nf">annotate</span><span class="p">(</span><span class="s">&quot;text&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">-5.5</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="o">=</span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="n">col</span><span class="o">=</span><span class="w"> </span><span class="s">&quot;blue&quot;</span><span class="p">,</span><span class="n">label</span><span class="o">=</span><span class="nf">paste0</span><span class="p">(</span><span class="s">&quot;n=&quot;</span><span class="p">,</span><span class="nf">length</span><span class="p">(</span><span class="n">downreg</span><span class="p">)))</span><span class="o">+</span>
<span class="w">    </span><span class="nf">annotate</span><span class="p">(</span><span class="s">&quot;text&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">5.5</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="o">=</span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="n">col</span><span class="o">=</span><span class="w"> </span><span class="s">&quot;red&quot;</span><span class="p">,</span><span class="n">label</span><span class="o">=</span><span class="nf">paste0</span><span class="p">(</span><span class="s">&quot;n=&quot;</span><span class="p">,</span><span class="nf">length</span><span class="p">(</span><span class="n">upreg</span><span class="p">)))</span>

<span class="w">  </span><span class="nf">ggsave</span><span class="p">(</span><span class="n">filename</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="s">&quot;volcano&quot;</span><span class="p">,</span><span class="w"> </span><span class="nf">names</span><span class="p">(</span><span class="n">coefs</span><span class="p">[</span><span class="n">i</span><span class="p">]),</span><span class="w"> </span><span class="s">&quot;.png&quot;</span><span class="p">),</span>
<span class="w">         </span><span class="n">path</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;figures/volcanoes/&quot;</span><span class="p">,</span>
<span class="w">         </span><span class="n">device</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;png&quot;</span><span class="p">,</span>
<span class="w">         </span><span class="n">width</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">640</span><span class="p">,</span>
<span class="w">         </span><span class="n">height</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">660</span><span class="p">,</span>
<span class="w">         </span><span class="n">units</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;px&quot;</span><span class="p">,</span>
<span class="w">         </span><span class="n">dpi</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">100</span><span class="p">)</span>
<span class="p">}</span>
</pre></div>
</div>
</li>
<li><p>ANOVA filtering</p>
<p>For multiple comparisons it is best to check if removing low
variance proteins results in better clustering/differentiation -
e.g. with heatmap/dendrogram and/or PCA</p>
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># drop &#39;coef&#39; argument for anova</span>
<span class="n">anova</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">topTable</span><span class="p">(</span><span class="n">fit_bayes</span><span class="p">,</span><span class="w"> </span><span class="n">adjust</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;BH&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">genelist</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">rownames</span><span class="p">(</span><span class="n">mat</span><span class="p">),</span><span class="w"> </span><span class="n">resort.by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;adj.P.val&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">number</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">nrow</span><span class="p">(</span><span class="n">qnorm</span><span class="p">))</span>

<span class="n">anova_filtr</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">anova</span><span class="o">$</span><span class="n">ProbeID</span><span class="p">[</span><span class="n">anova</span><span class="o">$</span><span class="n">adj.P.Val</span><span class="w"> </span><span class="o">&lt;=</span><span class="w"> </span><span class="m">0.05</span><span class="p">]</span>
<span class="n">temp</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">qnorm</span><span class="p">,</span><span class="w"> </span><span class="o">!</span><span class="n">Gene</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">anova_filtr</span><span class="p">)</span>

<span class="n">pheatmap</span><span class="o">::</span><span class="nf">pheatmap</span><span class="p">(</span><span class="n">temp</span><span class="p">[</span><span class="m">1</span><span class="o">:</span><span class="m">18</span><span class="p">],</span>
<span class="w">                   </span><span class="n">scale</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;row&quot;</span><span class="p">,</span>
<span class="w">                   </span><span class="n">show_rownames</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span>
<span class="w">                   </span><span class="n">cutree_rows</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">5</span><span class="p">,</span>
<span class="w">                   </span><span class="n">clustering_distance_cols</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;euclidean&quot;</span><span class="p">,</span>
<span class="w">                   </span><span class="n">clustering_distance_rows</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;correlation&quot;</span><span class="p">,</span>
<span class="w">                   </span><span class="n">annotation_col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">annot_col</span><span class="p">,</span>
<span class="w">                   </span><span class="n">annotation_colors</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">annot_colors</span><span class="p">,</span>
<span class="w">                   </span><span class="n">main</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;ANOVA filtered&quot;</span><span class="p">)</span>
</pre></div>
</div>
</li>
<li><p>Visualize terms on a heatmap</p>
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">term</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;negative regulation of cell cycle process&quot;</span>
<span class="n">filtr</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">str_split_1</span><span class="p">(</span><span class="n">gsea_results</span><span class="o">$</span><span class="n">core_enrichment</span><span class="p">[</span><span class="n">gsea_results</span><span class="o">$</span><span class="n">Description</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="n">term</span><span class="p">],</span><span class="w"> </span><span class="n">pattern</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;/&quot;</span><span class="p">)</span>

<span class="n">pheatmap</span><span class="o">::</span><span class="nf">pheatmap</span><span class="p">(</span><span class="n">qnorm</span><span class="p">[</span><span class="n">qnorm</span><span class="o">$</span><span class="n">Gene</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">filtr</span><span class="p">,</span><span class="m">1</span><span class="o">:</span><span class="m">18</span><span class="p">],</span>
<span class="w">                   </span><span class="n">scale</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;row&quot;</span><span class="p">,</span>
<span class="w">                   </span><span class="n">show_rownames</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span>
<span class="w">                   </span><span class="n">clustering_distance_cols</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;euclidean&quot;</span><span class="p">,</span>
<span class="w">                   </span><span class="n">clustering_distance_rows</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;correlation&quot;</span><span class="p">,</span>
<span class="w">                   </span><span class="n">annotation_col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">annot_col</span><span class="p">,</span>
<span class="w">                   </span><span class="n">annotation_colors</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">annot_colors</span><span class="p">,</span>
<span class="w">                   </span><span class="n">main</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">term</span><span class="p">)</span>
</pre></div>
</div>
</li>
</ul>
</li>
</ul>
</section>
<section id="pheatmap">
<h2>4. pheatmap<a class="headerlink" href="#pheatmap" title="Permalink to this heading">#</a></h2>
<p>Global heatmap with experimental groups as column annotation.</p>
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1"># Tidying up--------------------------------------------------------------------</span>
<span class="c1">#drop nas and set rownames</span>
<span class="n">msqrob</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">drop_na</span><span class="p">(</span><span class="n">msqrob</span><span class="p">)</span>
<span class="n">msqrob</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">column_to_rownames</span><span class="p">(</span><span class="n">msqrob</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;Gene&quot;</span><span class="p">)</span>
<span class="n">msqrob</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">as.matrix</span><span class="p">(</span><span class="n">msqrob</span><span class="p">[</span><span class="m">1</span><span class="o">:</span><span class="m">18</span><span class="p">])</span>

<span class="c1"># Global heatmap-----------------------------------------------------------------------</span>
<span class="c1">#Column annotation - each column is a named grouping variable</span>
<span class="n">annot_col</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">data.frame</span><span class="p">(</span>
<span class="w">  </span><span class="n">Condition.L1</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">as.factor</span><span class="p">(</span>
<span class="w">    </span><span class="nf">c</span><span class="p">(</span><span class="nf">rep</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;2D&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;3D.young&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;3D.old&quot;</span><span class="p">),</span><span class="w"> </span><span class="n">each</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">4</span><span class="p">),</span>
<span class="w">      </span><span class="nf">rep</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;PDX.2D&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;PDX.3D&quot;</span><span class="p">),</span><span class="w"> </span><span class="n">each</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">3</span><span class="p">))),</span>
<span class="w">  </span><span class="n">Condition.L2</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">as.factor</span><span class="p">(</span>
<span class="w">    </span><span class="nf">c</span><span class="p">(</span><span class="nf">rep</span><span class="p">(</span><span class="s">&quot;2D&quot;</span><span class="p">,</span><span class="w"> </span><span class="m">4</span><span class="p">),</span><span class="w"> </span>
<span class="w">      </span><span class="nf">rep</span><span class="p">(</span><span class="s">&quot;3D&quot;</span><span class="p">,</span><span class="w"> </span><span class="m">8</span><span class="p">),</span><span class="w"> </span>
<span class="w">      </span><span class="nf">rep</span><span class="p">(</span><span class="s">&quot;PDX&quot;</span><span class="p">,</span><span class="w"> </span><span class="m">6</span><span class="p">))))</span>
<span class="c1">#set sample names as rownames</span>
<span class="nf">row.names</span><span class="p">(</span><span class="n">annot_col</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">colnames</span><span class="p">(</span><span class="n">msqrob</span><span class="p">)[</span><span class="m">1</span><span class="o">:</span><span class="m">18</span><span class="p">]</span>

<span class="c1">#Colors mapping for column annotation</span>
<span class="c1">#A named list for each grouping with color values</span>
<span class="nf">library</span><span class="p">(</span><span class="n">RColorBrewer</span><span class="p">)</span>
<span class="n">annot_colors</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">list</span><span class="p">(</span><span class="n">Condition.L1</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">rev</span><span class="p">(</span><span class="nf">brewer.pal</span><span class="p">(</span><span class="m">5</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;Paired&quot;</span><span class="p">)),</span>
<span class="w">                     </span><span class="n">Condition.L2</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;firebrick1&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;darkgreen&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;blue&quot;</span><span class="p">))</span>
<span class="c1">#Map names of experimental groups to the colors</span>
<span class="nf">names</span><span class="p">(</span><span class="n">annot_colors</span><span class="o">$</span><span class="n">Condition.L1</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">levels</span><span class="p">(</span><span class="n">annot_col</span><span class="o">$</span><span class="n">Condition.L1</span><span class="p">)</span>
<span class="nf">names</span><span class="p">(</span><span class="n">annot_colors</span><span class="o">$</span><span class="n">Condition.L2</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">levels</span><span class="p">(</span><span class="n">annot_col</span><span class="o">$</span><span class="n">Condition.L2</span><span class="p">)</span>

<span class="c1">#Plot hm</span>
<span class="n">hm</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">pheatmap</span><span class="o">::</span><span class="nf">pheatmap</span><span class="p">(</span><span class="n">msqrob</span><span class="p">[</span><span class="m">1</span><span class="o">:</span><span class="m">18</span><span class="p">],</span>
<span class="w">           </span><span class="n">scale</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;row&quot;</span><span class="p">,</span>
<span class="w">           </span><span class="n">show_rownames</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span>
<span class="w">           </span><span class="n">clustering_distance_cols</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;euclidean&quot;</span><span class="p">,</span>
<span class="w">           </span><span class="n">clustering_distance_rows</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;correlation&quot;</span><span class="p">,</span>
<span class="w">           </span><span class="n">annotation_col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">annot_col</span><span class="p">,</span>
<span class="w">           </span><span class="n">annotation_colors</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">annot_colors</span><span class="p">,</span>
<span class="w">           </span><span class="n">main</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Global heatmap&quot;</span><span class="p">)</span>
</pre></div>
</div>
<p>I like to check clustering after ANOVA filtering &lt;0.05 adj.p-value and
then after DE (limma pkg below).</p>
<p>Explore row dendrogram of hm after filtering of matrix.</p>
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="n">filtr_deps</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">unlist</span><span class="p">(</span><span class="n">deps_list</span><span class="p">)</span><span class="w"> </span><span class="o">|&gt;</span><span class="w"> </span><span class="nf">unique</span><span class="p">()</span>

<span class="c1"># View row tree, decide on nr of clusters</span>
<span class="n">hm</span><span class="o">$</span><span class="n">tree_row</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">  </span><span class="nf">as.dendrogram</span><span class="p">()</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">plot</span><span class="p">(</span><span class="n">horiz</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span>

<span class="c1"># assign genes to clusters and annotate in hm, extract and explore in string network/go</span>
<span class="n">clusters</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">cutree</span><span class="p">(</span><span class="n">hm</span><span class="o">$</span><span class="n">tree_row</span><span class="p">,</span><span class="w"> </span><span class="n">k</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">5</span><span class="p">)</span>
<span class="n">clusters</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">as.data.frame</span><span class="p">(</span><span class="n">clusters</span><span class="p">)</span><span class="w"> </span><span class="o">|&gt;</span><span class="w"> </span><span class="nf">mutate</span><span class="p">(</span><span class="n">clusters</span><span class="o">=</span><span class="nf">as.factor</span><span class="p">(</span><span class="n">clusters</span><span class="p">))</span>
<span class="n">clipr</span><span class="o">::</span><span class="nf">write_clip</span><span class="p">(</span><span class="nf">rownames</span><span class="p">(</span><span class="n">clusters</span><span class="p">)[</span><span class="n">clusters</span><span class="o">$</span><span class="n">clusters</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="m">5</span><span class="p">])</span>

<span class="c1">#plot hm again with clustered rows separated</span>
<span class="n">hm</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">pheatmap</span><span class="o">::</span><span class="nf">pheatmap</span><span class="p">(</span><span class="n">qnorm</span><span class="p">[</span><span class="n">qnorm</span><span class="o">$</span><span class="n">Gene</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">filtr_deps</span><span class="p">,</span><span class="m">1</span><span class="o">:</span><span class="m">18</span><span class="p">],</span>
<span class="w">                   </span><span class="n">scale</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;row&quot;</span><span class="p">,</span>
<span class="w">                   </span><span class="n">show_rownames</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span>
<span class="w">                   </span><span class="n">cutree_rows</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">5</span><span class="p">,</span>
<span class="w">                   </span><span class="n">annotation_row</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">clusters</span><span class="p">,</span>
<span class="w">                   </span><span class="n">clustering_distance_cols</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;euclidean&quot;</span><span class="p">,</span>
<span class="w">                   </span><span class="n">clustering_distance_rows</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;correlation&quot;</span><span class="p">,</span>
<span class="w">                   </span><span class="n">annotation_col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">annot_col</span><span class="p">,</span>
<span class="w">                   </span><span class="n">annotation_colors</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">annot_colors</span><span class="p">,</span>
<span class="w">                   </span><span class="n">main</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Significant DEPs&quot;</span><span class="p">)</span>
</pre></div>
</div>
<p>Full list of all DEPs across comparisons:</p>
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1">#create deps list if multiple comparisons</span>
<span class="n">deps_list</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="kr">function</span><span class="p">(</span><span class="n">fit</span><span class="p">){</span>
<span class="w">  </span><span class="n">comb</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">list</span><span class="p">()</span>
<span class="w">  </span><span class="kr">for</span><span class="w"> </span><span class="p">(</span><span class="n">i</span><span class="w"> </span><span class="kr">in</span><span class="w"> </span><span class="n">coefs</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">    </span><span class="n">deps</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">topTable</span><span class="p">(</span><span class="n">fit_bayes</span><span class="p">,</span><span class="w"> </span><span class="n">coef</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">i</span><span class="p">,</span><span class="w"> </span><span class="n">adjust</span><span class="o">=</span><span class="s">&quot;BH&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">genelist</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">rownames</span><span class="p">(</span><span class="n">mat</span><span class="p">),</span><span class="w"> </span><span class="n">resort.by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;logFC&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">number</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">nrow</span><span class="p">(</span><span class="n">qnorm</span><span class="p">))</span>
<span class="w">    </span>
<span class="w">    </span><span class="n">volc_data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">deps</span>
<span class="w">    </span><span class="n">volc_data</span><span class="o">$</span><span class="n">Category</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&quot;Not significant&quot;</span>
<span class="w">    </span><span class="n">volc_data</span><span class="o">$</span><span class="n">Category</span><span class="p">[</span><span class="n">volc_data</span><span class="o">$</span><span class="n">logFC</span><span class="w"> </span><span class="o">&lt;=</span><span class="w"> </span><span class="m">-1.3</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="n">volc_data</span><span class="o">$</span><span class="n">adj.P.Val</span><span class="w"> </span><span class="o">&lt;=</span><span class="w"> </span><span class="m">0.05</span><span class="p">]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&quot;Down-regulated&quot;</span>
<span class="w">    </span><span class="n">volc_data</span><span class="o">$</span><span class="n">Category</span><span class="p">[</span><span class="n">volc_data</span><span class="o">$</span><span class="n">logFC</span><span class="w"> </span><span class="o">&gt;=</span><span class="w"> </span><span class="m">1.3</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="n">volc_data</span><span class="o">$</span><span class="n">adj.P.Val</span><span class="w"> </span><span class="o">&lt;=</span><span class="w"> </span><span class="m">0.05</span><span class="p">]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&quot;Up-regulated&quot;</span>
<span class="w">    </span><span class="n">volc_data</span><span class="o">$</span><span class="n">Label</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="kc">NA</span>
<span class="w">    </span><span class="n">volc_data</span><span class="o">$</span><span class="n">Label</span><span class="p">[</span><span class="n">volc_data</span><span class="o">$</span><span class="n">Category</span><span class="w"> </span><span class="o">!=</span><span class="w"> </span><span class="s">&quot;Not significant&quot;</span><span class="p">]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">volc_data</span><span class="o">$</span><span class="n">ID</span><span class="p">[</span><span class="n">volc_data</span><span class="o">$</span><span class="n">Category</span><span class="w"> </span><span class="o">!=</span><span class="w"> </span><span class="s">&quot;Not significant&quot;</span><span class="p">]</span>
<span class="w">    </span>
<span class="w">    </span><span class="n">upreg</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">volc_data</span><span class="o">$</span><span class="n">Label</span><span class="p">[</span><span class="n">volc_data</span><span class="o">$</span><span class="n">Category</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s">&quot;Up-regulated&quot;</span><span class="p">]</span>
<span class="w">    </span><span class="n">downreg</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">volc_data</span><span class="o">$</span><span class="n">Label</span><span class="p">[</span><span class="n">volc_data</span><span class="o">$</span><span class="n">Category</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s">&quot;Down-regulated&quot;</span><span class="p">]</span>
<span class="w">    </span>
<span class="w">    </span><span class="n">comb</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">append</span><span class="p">(</span><span class="n">comb</span><span class="p">,</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="nf">list</span><span class="p">(</span><span class="n">upreg</span><span class="p">),</span><span class="w"> </span><span class="nf">list</span><span class="p">(</span><span class="n">downreg</span><span class="p">)))</span>
<span class="w">  </span><span class="p">}</span>
<span class="w">  </span><span class="n">comb_names</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">rep</span><span class="p">((</span><span class="nf">names</span><span class="p">(</span><span class="n">coefs</span><span class="p">)),</span><span class="n">each</span><span class="o">=</span><span class="m">2</span><span class="p">,</span><span class="n">times</span><span class="o">=</span><span class="m">1</span><span class="p">)</span><span class="w"> </span><span class="o">|&gt;</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="nf">rep</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;_upreg&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;_downreg&quot;</span><span class="p">),</span><span class="w"> </span><span class="n">every</span><span class="o">=</span><span class="m">2</span><span class="p">))</span>
<span class="w">  </span><span class="nf">names</span><span class="p">(</span><span class="n">comb</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">comb_names</span>
<span class="w">  </span><span class="kr">return</span><span class="p">(</span><span class="n">comb</span><span class="p">)</span>
<span class="p">}</span>
</pre></div>
</div>
</section>
<section id="pcatools">
<h2>5. PCAtools<a class="headerlink" href="#pcatools" title="Permalink to this heading">#</a></h2>
<p>Principal component analysis. Better results after removing low variance
features (e.g. after ANOVA).</p>
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">library</span><span class="p">(</span><span class="n">PCAtools</span><span class="p">)</span>
<span class="c1">#column to rownames for pca function</span>
<span class="n">mat</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">as.matrix</span><span class="p">(</span><span class="n">msqrob</span><span class="p">[</span><span class="m">1</span><span class="o">:</span><span class="m">18</span><span class="p">])</span>
<span class="c1">#pca</span>
<span class="n">p</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">pca</span><span class="p">(</span><span class="n">mat</span><span class="p">,</span><span class="w"> </span><span class="n">metadata</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">annot_col</span><span class="p">,</span><span class="w"> </span><span class="n">scale</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span><span class="w"> </span><span class="c1">#annot_col same as in pheatmap</span>

<span class="c1">#plot PCA</span>
<span class="nf">biplot</span><span class="p">(</span><span class="n">p</span><span class="p">,</span><span class="w"> </span>
<span class="w">       </span><span class="n">lab</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">colnames</span><span class="p">(</span><span class="n">mat</span><span class="p">),</span><span class="w"> </span>
<span class="w">       </span><span class="n">labSize</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">5</span><span class="p">,</span><span class="w"> </span>
<span class="w">       </span><span class="n">showLoadings</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span><span class="w"> </span><span class="c1">#show genes driving the variation</span>
<span class="w">       </span><span class="n">boxedLoadingsNames</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span><span class="w"> </span>
<span class="w">       </span><span class="n">colby</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Condition.L1&quot;</span><span class="p">,</span><span class="w"> </span><span class="c1">#colors of points</span>
<span class="w">       </span><span class="n">colkey</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">annot_colors</span><span class="o">$</span><span class="n">Condition.L1</span><span class="p">,</span><span class="w"> </span><span class="c1">#names of points, same as in pheatmap</span>
<span class="w">       </span><span class="n">title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;This is PCA biplot&quot;</span><span class="p">,</span>
<span class="w">       </span><span class="n">encircle</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="c1">#encircle groups (no stats applied - for that use &#39;eclipse&#39; arg)</span>
<span class="w">       </span><span class="n">hline</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="c1">#add 0,0 coordinates</span>
<span class="w">       </span><span class="n">vline</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0</span><span class="p">)</span>

<span class="c1">#show screeplot</span>
<span class="nf">screeplot</span><span class="p">(</span><span class="n">p</span><span class="p">)</span>

<span class="c1">#loadings plot to screen genes driving the variation in components</span>
<span class="nf">plotloadings</span><span class="p">(</span><span class="n">p</span><span class="p">)</span>
</pre></div>
</div>
</section>
<section id="eulerr">
<h2>6. eulerr<a class="headerlink" href="#eulerr" title="Permalink to this heading">#</a></h2>
<p>Create area-proportional venn/euler diagrams. Most frequently used for
peptide/protein overlap between samples.</p>
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1">#create a named list with peptide/proteins ids for each group/sample/comparison</span>
<span class="n">comb</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">list</span><span class="p">(</span><span class="n">l1</span><span class="p">,</span><span class="w"> </span><span class="n">l2</span><span class="p">,</span><span class="w"> </span><span class="n">l3</span><span class="p">,</span><span class="w"> </span><span class="n">l4</span><span class="p">)</span>
<span class="nf">names</span><span class="p">(</span><span class="n">comb</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;Sample1&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;Sample2&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;Sample3&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;Sample4&quot;</span><span class="p">)</span>

<span class="nf">euler</span><span class="p">(</span><span class="n">comb</span><span class="p">)</span><span class="w"> </span><span class="o">|&gt;</span><span class="w"> </span><span class="nf">plot</span><span class="p">(</span><span class="n">quantities</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="c1">#area-proportional?</span>
<span class="w">                    </span><span class="n">main</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;title&quot;</span><span class="p">,</span><span class="w"> </span>
<span class="w">                    </span><span class="n">fills</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;#AB3428&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;#FFB703&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;white&quot;</span><span class="p">))</span><span class="w"> </span><span class="c1">#circle colors</span>
</pre></div>
</div>
</section>
<section id="corrplot">
<h2>7. corrplot<a class="headerlink" href="#corrplot" title="Permalink to this heading">#</a></h2>
<p>Visualize correlation matrix in a heatmap. Assess pre- and
post-normalization.</p>
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="c1">#compute correlation coefs on matrix w/o NAs</span>
<span class="n">corr</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">cor</span><span class="p">(</span><span class="nf">log2</span><span class="p">(</span><span class="n">mat</span><span class="p">),</span><span class="w"> </span><span class="n">use</span><span class="o">=</span><span class="s">&quot;complete.obs&quot;</span><span class="p">)</span>
<span class="c1">#visualize</span>
<span class="nf">corrplot</span><span class="p">(</span><span class="n">corr</span><span class="p">,</span><span class="w"> </span><span class="n">method</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;color&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">hclust.method</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;ward.D2&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">is.corr</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span><span class="w"> </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">COL1</span><span class="p">(</span><span class="s">&quot;OrRd&quot;</span><span class="p">))</span>
</pre></div>
</div>
</section>
</section>
<section class="tex2jax_ignore mathjax_ignore" id="session-info">
<h1>Session info<a class="headerlink" href="#session-info" title="Permalink to this heading">#</a></h1>
<div class="highlight-r notranslate"><div class="highlight"><pre><span></span><span class="nf">sessionInfo</span><span class="p">()</span>
</pre></div>
</div>
<div class="highlight-none notranslate"><div class="highlight"><pre><span></span>## R version 4.3.2 (2023-10-31 ucrt)
## Platform: x86_64-w64-mingw32/x64 (64-bit)
## Running under: Windows 10 x64 (build 19045)
## 
## Matrix products: default
## 
## 
## locale:
## [1] LC_COLLATE=Polish_Poland.utf8  LC_CTYPE=Polish_Poland.utf8   
## [3] LC_MONETARY=Polish_Poland.utf8 LC_NUMERIC=C                  
## [5] LC_TIME=Polish_Poland.utf8    
## 
## time zone: Europe/Warsaw
## tzcode source: internal
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
##  [1] corrplot_0.92              eulerr_7.0.0              
##  [3] PCAtools_2.14.0            ggrepel_0.9.5             
##  [5] pheatmap_1.0.12            enrichplot_1.22.0         
##  [7] clusterProfiler_4.11.0.001 limma_3.58.1              
##  [9] affy_1.80.0                Biobase_2.62.0            
## [11] BiocGenerics_0.48.1        lubridate_1.9.3           
## [13] forcats_1.0.0              stringr_1.5.1             
## [15] dplyr_1.1.4                purrr_1.0.2               
## [17] readr_2.1.5                tidyr_1.3.1               
## [19] tibble_3.2.1               ggplot2_3.4.4             
## [21] tidyverse_2.0.0           
## 
## loaded via a namespace (and not attached):
##   [1] RColorBrewer_1.1-3        rstudioapi_0.15.0        
##   [3] jsonlite_1.8.8            magrittr_2.0.3           
##   [5] farver_2.1.1              rmarkdown_2.25           
##   [7] fs_1.6.3                  zlibbioc_1.48.0          
##   [9] vctrs_0.6.5               DelayedMatrixStats_1.24.0
##  [11] memoise_2.0.1             RCurl_1.98-1.14          
##  [13] ggtree_3.10.0             S4Arrays_1.2.0           
##  [15] htmltools_0.5.7           SparseArray_1.2.3        
##  [17] gridGraphics_0.5-1        plyr_1.8.9               
##  [19] cachem_1.0.8              igraph_2.0.1.1           
##  [21] lifecycle_1.0.4           pkgconfig_2.0.3          
##  [23] rsvd_1.0.5                Matrix_1.6-5             
##  [25] R6_2.5.1                  fastmap_1.1.1            
##  [27] gson_0.1.0                GenomeInfoDbData_1.2.11  
##  [29] MatrixGenerics_1.14.0     digest_0.6.34            
##  [31] aplot_0.2.2               colorspace_2.1-0         
##  [33] patchwork_1.2.0           AnnotationDbi_1.64.1     
##  [35] S4Vectors_0.40.2          dqrng_0.3.2              
##  [37] irlba_2.3.5.1             RSQLite_2.3.5            
##  [39] beachmat_2.18.0           fansi_1.0.6              
##  [41] timechange_0.3.0          abind_1.4-5              
##  [43] httr_1.4.7                polyclip_1.10-6          
##  [45] compiler_4.3.2            bit64_4.0.5              
##  [47] withr_3.0.0               BiocParallel_1.36.0      
##  [49] viridis_0.6.5             DBI_1.2.1                
##  [51] ggforce_0.4.1             MASS_7.3-60.0.1          
##  [53] DelayedArray_0.28.0       HDO.db_0.99.1            
##  [55] tools_4.3.2               ape_5.7-1                
##  [57] scatterpie_0.2.1          glue_1.7.0               
##  [59] nlme_3.1-164              GOSemSim_2.28.1          
##  [61] grid_4.3.2                shadowtext_0.1.3         
##  [63] reshape2_1.4.4            fgsea_1.28.0             
##  [65] generics_0.1.3            gtable_0.3.4             
##  [67] tzdb_0.4.0                preprocessCore_1.64.0    
##  [69] data.table_1.15.0         hms_1.1.3                
##  [71] ScaledMatrix_1.10.0       BiocSingular_1.18.0      
##  [73] tidygraph_1.3.1           utf8_1.2.4               
##  [75] XVector_0.42.0            pillar_1.9.0             
##  [77] yulab.utils_0.1.4         splines_4.3.2            
##  [79] tweenr_2.0.2              treeio_1.26.0            
##  [81] lattice_0.22-5            bit_4.0.5                
##  [83] tidyselect_1.2.0          GO.db_3.18.0             
##  [85] Biostrings_2.70.2         knitr_1.45               
##  [87] gridExtra_2.3             IRanges_2.36.0           
##  [89] stats4_4.3.2              xfun_0.41                
##  [91] graphlayouts_1.1.0        statmod_1.5.0            
##  [93] matrixStats_1.2.0         stringi_1.8.3            
##  [95] lazyeval_0.2.2            ggfun_0.1.4              
##  [97] yaml_2.3.8                evaluate_0.23            
##  [99] codetools_0.2-19          ggraph_2.1.0             
## [101] qvalue_2.34.0             BiocManager_1.30.22      
## [103] ggplotify_0.1.2           cli_3.6.2                
## [105] affyio_1.72.0             munsell_0.5.0            
## [107] Rcpp_1.0.12               GenomeInfoDb_1.38.5      
## [109] png_0.1-8                 parallel_4.3.2           
## [111] blob_1.2.4                DOSE_3.28.2              
## [113] sparseMatrixStats_1.14.0  bitops_1.0-7             
## [115] viridisLite_0.4.2         tidytree_0.4.6           
## [117] scales_1.3.0              crayon_1.5.2             
## [119] rlang_1.1.3               cowplot_1.1.3            
## [121] fastmatch_1.1-4           KEGGREST_1.42.0
</pre></div>
</div>
</section>
<section class="tex2jax_ignore mathjax_ignore" id="more-sources">
<h1>More sources<a class="headerlink" href="#more-sources" title="Permalink to this heading">#</a></h1>
<ul class="simple">
<li><p><a class="reference external" href="https://bioconductor.org/packages/release/bioc/vignettes/limma/inst/doc/usersguide.pdf">https://bioconductor.org/packages/release/bioc/vignettes/limma/inst/doc/usersguide.pdf</a></p></li>
<li><p><a class="reference external" href="https://yulab-smu.top/biomedical-knowledge-mining-book/index.html">https://yulab-smu.top/biomedical-knowledge-mining-book/index.html</a></p></li>
<li><p><a class="reference external" href="https://davetang.org/muse/2018/05/15/making-a-heatmap-in-r-with-the-pheatmap-package/">https://davetang.org/muse/2018/05/15/making-a-heatmap-in-r-with-the-pheatmap-package/</a></p></li>
</ul>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            name: "python3",
            path: "./."
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

                </article>
              

              
              
              
              
                <footer class="prev-next-footer d-print-none">
                  
<div class="prev-next-area">
    <a class="left-prev"
       href="git.html"
       title="previous page">
      <i class="fa-solid fa-angle-left"></i>
      <div class="prev-next-info">
        <p class="prev-next-subtitle">previous</p>
        <p class="prev-next-title">Git &amp; GitHub</p>
      </div>
    </a>
    <a class="right-next"
       href="setup.html"
       title="next page">
      <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">System setup</p>
      </div>
      <i class="fa-solid fa-angle-right"></i>
    </a>
</div>
                </footer>
              
            </div>
            
            
              
                <div class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">


  <div class="sidebar-secondary-item">
  <div class="page-toc tocsection onthispage">
    <i class="fa-solid fa-list"></i> Contents
  </div>
  <nav class="bd-toc-nav page-toc">
    <ul class="visible nav section-nav flex-column">
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#">R stack for -omics data</a></li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#scope">Scope</a><ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#affy">1. affy</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#limma">2. limma</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#clusterprofiler">3. clusterProfiler</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#pheatmap">4. pheatmap</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#pcatools">5. PCAtools</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#eulerr">6. eulerr</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#corrplot">7. corrplot</a></li>
</ul>
</li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#session-info">Session info</a></li>
<li class="toc-h1 nav-item toc-entry"><a class="reference internal nav-link" href="#more-sources">More sources</a></li>
</ul>

  </nav></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
<div class="bd-footer-content__inner container">
  
  <div class="footer-item">
    
<p class="component-author">
By Martyna Muszczek
</p>

  </div>
  
  <div class="footer-item">
    

  <p class="copyright">
    
      © Copyright 2023.
      <br/>
    
  </p>

  </div>
  
  <div class="footer-item">
    
  </div>
  
  <div class="footer-item">
    
  </div>
  
</div>
          </footer>
        

      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b"></script>
<script src="_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b"></script>

  <footer class="bd-footer">
  </footer>
  </body>
</html>