Search.setIndex({"docnames": ["R_stack_cheatsheet", "Week1", "Week2-workflows", "bfb", "conda_cheatsheet", "git", "intro", "jupyterlabtrick_wsl", "proteogenomics", "setup"], "filenames": ["R_stack_cheatsheet.md", "Week1.ipynb", "Week2-workflows.ipynb", "bfb.ipynb", "conda_cheatsheet.ipynb", "git.ipynb", "intro.md", "jupyterlabtrick_wsl.ipynb", "proteogenomics.md", "setup.ipynb"], "titles": ["R stack for -omics data", "Week 1 - setup, sequencing QC, mapping, variant calling", "Week 2 - workflows with Nextflow", "Course notes: Bioinformatics for Biologists", "Conda", "Git &amp; GitHub", "Bioinformatics cheatsheets", "WSL incompatibility solutions", "Proteogenomics for neoantigen discovery", "System setup"], "terms": {"martyna": [0, 8], "siewiera": [0, 8], "2024": 0, "02": [0, 4], "14": 0, "thi": [0, 1, 3, 4, 5, 7, 8, 9], "i": [0, 1, 3, 4, 5, 9], "quick": [0, 6], "overview": 0, "my": [0, 1, 9], "current": [0, 5, 6], "us": [0, 1, 5, 6, 8, 9], "proteom": [0, 6], "analysi": [0, 6], "The": [0, 4, 6], "aim": 0, "cheatsheet": [0, 1], "organ": [0, 1, 6], "packag": [0, 1, 6, 8, 9], "standard": 0, "easi": 0, "get": [0, 9], "lost": 0, "all": [0, 4, 5], "avail": [0, 4], "bioconductor": 0, "end": [0, 8], "don": 0, "t": [0, 8, 9], "work": [0, 1, 4, 8, 9], "well": 0, "ar": [0, 4, 8], "miss": [0, 9], "lot": 0, "function": [0, 6], "need": [0, 1, 4, 5, 9], "combin": [0, 8], "anoth": [0, 4], "what": [0, 5, 9], "ha": 0, "me": 0, "so": [0, 1, 4, 5, 9], "far": 0, "want": [0, 8], "place": [0, 6], "where": [0, 8], "everyth": 0, "sum": 0, "up": [0, 1, 6], "somewhat": 0, "sequenti": 0, "order": 0, "have": [0, 1, 2, 9], "been": 0, "mainli": 0, "two": [0, 1], "three": 0, "experiment": 0, "group": 0, "wa": [0, 6], "an": [0, 1, 4, 5], "instanc": [0, 5], "had": [0, 9], "creat": [0, 1, 5, 6, 9], "long": 0, "just": [0, 1, 5], "automat": [0, 9], "de": 0, "term": 0, "enrich": 0, "visual": [0, 1], "multipl": 0, "These": 0, "can": [0, 1, 4, 5, 8, 9], "found": [0, 9], "under": 0, "test": [0, 9], "normal": [0, 8], "perform": 0, "ma": 0, "plot": 0, "blue": 0, "line": [0, 4, 5, 9], "repres": 0, "theoret": 0, "perfect": 0, "loess": 0, "curv": 0, "red": 0, "actual": 0, "similar": 0, "thei": 0, "better": [0, 8], "note": [0, 6, 9], "principl": 0, "major": 0, "featur": [0, 9], "protein": 0, "gene": 0, "do": [0, 1, 4, 7, 9], "chang": [0, 1, 2, 5, 9], "express": 0, "most": [0, 1, 6, 9], "condit": 0, "trend": 0, "sampl": 0, "m": [0, 5], "log2": 0, "x": [0, 4], "y": [0, 9], "aka": 0, "log2fc": 0, "A": [0, 8], "xy": 0, "mean": [0, 4], "averag": 0, "librari": [0, 8], "name": [0, 4, 5, 8], "list": [0, 8, 9], "column": 0, "index": [0, 1, 4, 8], "each": [0, 1, 4], "9": 0, "colnam": 0, "design": 0, "from": [0, 1, 3, 4, 5, 6, 8, 9], "pkg": [0, 4], "search": [0, 4], "below": [0, 6], "take": [0, 1], "matrix": 0, "rowmean": 0, "qnorm": 0, "c": [0, 1, 8, 9], "x3d": 0, "x2d": 0, "save": [0, 5, 8], "maplot": 0, "cex": 0, "titl": 0, "normalis": 0, "origin": [0, 5], "dedic": 0, "microarrai": 0, "rna": [0, 3, 6], "seq": [0, 3, 6], "which": [0, 3, 4, 5, 7, 9], "also": [0, 4], "method": 0, "differenti": 0, "load": [0, 1], "requir": [0, 8, 9], "tidyvers": [0, 8], "prepar": 0, "transform": 0, "tidi": 0, "mat": 0, "read": [0, 1], "delim": 0, "expr_matrix": 0, "tsv": 0, "head": [0, 5], "differ": [0, 4], "assess": 0, "boxplot": 0, "jitter": 0, "normalizevsn": 0, "normalizequantil": 0, "normalizecyclicloess": 0, "normalizemedianvalu": 0, "tabl": 0, "0": [0, 5], "assign": 0, "There": 0, "mani": [0, 8], "wai": [0, 1], "find": 0, "follow": [0, 1, 5, 7, 8, 9], "one": [0, 1, 4, 8], "easiest": 0, "In": [0, 4, 9], "model": 0, "rep": 0, "first": [0, 4, 8], "digit": 0, "fill": 0, "nr": [0, 1], "replic": 0, "indic": 0, "second": 0, "factor": 0, "ctrl": 0, "treat1": 0, "treat2": 0, "treat3": 0, "exp": 0, "manual": 0, "row": 0, "lm": 0, "fit": 0, "contrast": 0, "comparison": 0, "between": [0, 8], "made": 0, "To": [0, 1], "make": [0, 4, 5], "thing": 0, "clear": 0, "next": 0, "step": [0, 1, 9], "number": 0, "comment": 0, "coef": 0, "argument": 0, "later": [0, 1, 8], "lmfit": 0, "cont": 0, "makecontrast": 0, "ctrl_treat1": 0, "ctrl_treat2": 0, "ctrl_treat3": 0, "level": 0, "calcul": [0, 8], "bay": 0, "statist": 0, "fit_cont": 0, "fit_bay": 0, "ebay": 0, "write": 0, "down": 0, "whole": [0, 1, 8], "frame": 0, "taken": 0, "account": [0, 5], "see": [0, 6], "id": 0, "rownam": 0, "dep": [0, 4], "toptabl": 0, "adjust": 0, "bh": 0, "genelist": 0, "resort": 0, "logfc": 0, "nrow": 0, "For": [0, 4, 6, 9], "volcano": 0, "volcanoplot": 0, "highlight": 0, "10": [0, 8], "ablin": 0, "h": [0, 8], "v": [0, 1, 5, 8], "lty": 0, "much": 0, "prefer": [0, 9], "version": [0, 4, 9], "retain": 0, "categori": 0, "volc_data": 0, "Not": 0, "signific": 0, "adj": 0, "p": [0, 1, 8, 9], "val": 0, "05": 0, "regul": 0, "label": 0, "na": 0, "theme_set": 0, "theme_light": 0, "ggplot": 0, "ae": 0, "log10": 0, "color": 0, "geom_point": 0, "alpha": 0, "scale_color_manu": 0, "valu": 0, "grai": 0, "lab": 0, "fold": 0, "geom_hlin": 0, "yintercept": 0, "geom_vlin": 0, "xintercept": 0, "scale_x_continu": 0, "limit": [0, 9], "scale_y_continu": 0, "addit": [0, 4], "format": 0, "ad": [0, 5], "n": [0, 1, 8, 9], "upreg": 0, "downreg": 0, "theme": 0, "legend": 0, "posit": 0, "top": 0, "text": 0, "element_text": 0, "size": [0, 1], "15": [0, 8], "geom_text_repel": 0, "colour": 0, "black": 0, "max": 0, "overlap": 0, "annot": 0, "col": 0, "paste0": 0, "length": 0, "ontologi": 0, "kegg": 0, "reactom": 0, "wikipathwai": 0, "gsea": 0, "ora": 0, "veri": [0, 4], "simplifi": 0, "result": [0, 1, 2], "remov": [0, 4, 5], "redund": 0, "eas": 0, "interpret": [0, 1, 9], "enrichplot": 0, "import": [0, 4, 5, 6], "like": [0, 4, 8], "cooper": 0, "esp": 0, "come": 0, "gsekegg": 0, "resolv": 0, "problem": [0, 1, 7], "consid": 0, "sure": [0, 4], "you": [0, 2, 4, 5, 8, 9], "latest": [0, 4, 9], "biocmanag": 0, "updat": [0, 9], "org": [0, 1, 8], "db": 0, "depend": [0, 4], "etc": 0, "If": [0, 9], "still": 0, "remot": [0, 5], "install_github": 0, "yulab": 0, "smu": 0, "download": [0, 1, 4, 9], "dev": 0, "last": 0, "time": [0, 4, 8], "noth": 0, "intern": 0, "eg": [0, 4, 5], "pvalu": 0, "": [0, 1, 5, 8, 9], "fc": 0, "charact": 0, "symbol": 0, "sort": [0, 1, 8], "decreas": 0, "true": [0, 7, 8], "rank": 0, "some": [0, 8], "accept": [0, 1, 9], "input": [0, 2, 9], "case": [0, 5, 8, 9], "convert": 0, "entrezid": 0, "bitr": 0, "fromtyp": 0, "totyp": 0, "orgdb": 0, "genelist2": 0, "filter": 0, "out": [0, 8], "unmap": 0, "sometim": [0, 4], "extern": 0, "tool": [0, 1, 6], "webgestalt": 0, "tab": 0, "seper": [0, 4], "file": [0, 1, 4, 5, 7, 8, 9], "txt": [0, 1], "sep": 0, "quot": 0, "fals": 0, "go": [0, 8, 9], "gsego": 0, "ont": 0, "bp": 0, "cc": 0, "mf": 0, "mingssiz": 0, "50": 0, "min": [0, 8], "map": 0, "maxgssiz": 0, "100": [0, 8], "pvaluecutoff": 0, "verbos": 0, "keytyp": 0, "view": [0, 1, 8], "gsea_result": 0, "ne": 0, "possibl": [0, 5], "merg": 0, "through": [0, 5], "simp": 0, "simp_result": 0, "hsa": 0, "ncbi": [0, 1], "geneid": 0, "20": [0, 1, 8, 9], "500": 0, "kegg_result": 0, "back": 0, "readabl": 0, "setread": 0, "visualis": 0, "choos": 0, "barplot": 0, "dotplot": 0, "ridgeplot": 0, "umap": 0, "cnetplot": 0, "check": [0, 1, 5], "clustprofil": 0, "vignett": 0, "d": [0, 8], "nice": 0, "ggplot2": 0, "theme_minim": 0, "arrang": 0, "desc": 0, "mutat": 0, "descript": 0, "geom_col": 0, "setsiz": 0, "scale_fill_gradi": 0, "low": 0, "high": 0, "its": [0, 4], "int": 0, "ctrlvtrt1": 0, "ctrlvtrt2": 0, "trt1vtrt2": 0, "compute_gsea": 0, "figur": 0, "path_fig": 0, "gsego_result": 0, "gsego_": 0, "_result": 0, "unfilt": 0, "path_data": 0, "dir": [0, 5], "exist": [0, 5], "showwarn": 0, "recurs": 0, "filename_data": 0, "_": 0, "filename_fig": 0, "_top30_": 0, "png": 0, "defin": [0, 4], "gssize": 0, "how": [0, 1, 8], "per": 0, "should": [0, 1, 7, 9], "via": 0, "gosemsim": 0, "label_format": 0, "60": 0, "ggsave": 0, "filenam": 0, "devic": 0, "path": [0, 9], "width": 0, "800": 0, "height": 0, "780": 0, "unit": 0, "px": 0, "dpi": 0, "els": 0, "NOT": 0, "onli": [0, 4, 8], "notsimplifi": 0, "compute_keggsea": 0, "gsekegg_result": 0, "gsekegg_": 0, "top30_": 0, "enrich_go": 0, "enrichgo_result": 0, "enrich_": 0, "_sim_": 0, "_top30_sim_": 0, "extract": 0, "messag": 0, "upregul": 0, "downregul": 0, "go_enr": 0, "enrichgo": 0, "univers": 0, "padjustmethod": 0, "go_enr_result": 0, "_upreg": 0, "dec": 0, "go_enr_sim": 0, "showcategori": 0, "30": [0, 8], "No": 0, "_downreg": 0, "not_simplifi": 0, "_notsim_": 0, "_notsim_upreg": 0, "_notsim_downreg": 0, "comparison_stat": 0, "640": 0, "660": 0, "anova": 0, "best": [0, 4], "varianc": 0, "cluster": 0, "e": 0, "g": [0, 8], "heatmap": 0, "dendrogram": 0, "pca": 0, "drop": 0, "anova_filtr": 0, "probeid": 0, "temp": 0, "18": 0, "scale": 0, "show_rownam": 0, "cutree_row": 0, "clustering_distance_col": 0, "euclidean": 0, "clustering_distance_row": 0, "correl": 0, "annotation_col": 0, "annot_col": 0, "annotation_color": 0, "annot_color": 0, "main": [0, 5], "neg": 0, "cell": 0, "cycl": 0, "process": [0, 1, 3, 6, 8], "filtr": 0, "str_split_1": 0, "core_enrich": 0, "pattern": [0, 5], "global": 0, "set": [0, 5, 6, 7, 9], "msqrob": 0, "drop_na": 0, "column_to_rownam": 0, "variabl": 0, "l1": 0, "2d": 0, "3d": 0, "young": 0, "old": 0, "pdx": 0, "l2": 0, "rcolorbrew": 0, "rev": 0, "brewer": 0, "pal": 0, "pair": [0, 8], "firebrick1": 0, "darkgreen": 0, "hm": 0, "after": [0, 9], "explor": [0, 1], "filtr_dep": 0, "unlist": 0, "deps_list": 0, "uniqu": [0, 8], "tree": 0, "decid": 0, "tree_row": 0, "horiz": 0, "string": 0, "network": 0, "cutre": 0, "k": 0, "clipr": 0, "write_clip": 0, "again": [0, 1, 4, 7, 9], "separ": 0, "annotation_row": 0, "full": [0, 8], "across": 0, "comb": 0, "append": 0, "comb_nam": 0, "everi": [0, 4], "return": 0, "princip": 0, "compon": 0, "metadata": 0, "same": [0, 4], "biplot": 0, "labsiz": 0, "showload": 0, "show": 0, "drive": 0, "variat": 0, "boxedloadingsnam": 0, "colbi": 0, "point": 0, "colkei": 0, "encircl": 0, "stat": [0, 1], "appli": 0, "eclips": 0, "arg": [0, 8], "hline": 0, "add": [0, 5, 8, 9], "coordin": 0, "vline": 0, "screeplot": 0, "screen": 0, "plotload": 0, "area": 0, "proport": 0, "venn": 0, "euler": 0, "diagram": 0, "frequent": 0, "peptid": 0, "l3": 0, "l4": 0, "sample1": 0, "sample2": 0, "sample3": 0, "sample4": 0, "quantiti": 0, "ab3428": 0, "ffb703": 0, "white": 0, "circl": 0, "pre": 0, "post": 0, "comput": 0, "w": 0, "o": [0, 1, 8, 9], "corr": 0, "cor": 0, "complet": 0, "ob": 0, "hclust": 0, "ward": 0, "d2": 0, "col1": 0, "orrd": 0, "readi": [0, 5, 9], "publish": 0, "panel": 0, "sessioninfo": 0, "2023": 0, "31": 0, "ucrt": 0, "platform": [0, 2, 9], "x86_64": [0, 9], "w64": 0, "mingw32": 0, "x64": 0, "64": 0, "bit": 0, "run": [0, 1, 4, 8, 9], "window": [0, 8], "build": 0, "19045": 0, "product": 0, "default": [0, 9], "local": 0, "lc_collat": 0, "polish_poland": 0, "utf8": 0, "lc_ctype": 0, "lc_monetari": 0, "lc_numer": 0, "lc_time": 0, "zone": 0, "europ": 0, "warsaw": 0, "tzcode": 0, "attach": 0, "base": [0, 6, 8], "graphic": 0, "grdevic": 0, "util": 0, "dataset": [0, 1, 9], "other": [0, 4, 8], "corrplot_0": 0, "92": 0, "eulerr_7": 0, "pcatools_2": 0, "ggrepel_0": 0, "pheatmap_1": 0, "12": [0, 2, 8], "enrichplot_1": 0, "22": [0, 9], "clusterprofiler_4": 0, "11": [0, 9], "001": 0, "limma_3": 0, "58": 0, "affy_1": 0, "80": 0, "biobase_2": 0, "62": 0, "biocgenerics_0": 0, "48": 0, "lubridate_1": 0, "13": 0, "forcats_1": 0, "stringr_1": 0, "dplyr_1": 0, "purrr_1": 0, "17": 0, "readr_2": 0, "tidyr_1": 0, "19": 0, "tibble_3": 0, "ggplot2_3": 0, "21": 0, "tidyverse_2": 0, "namespac": 0, "rcolorbrewer_1": 0, "rstudioapi_0": 0, "jsonlite_1": 0, "magrittr_2": 0, "farver_2": 0, "rmarkdown_2": 0, "25": 0, "fs_1": 0, "zlibbioc_1": 0, "vctrs_0": 0, "delayedmatrixstats_1": 0, "24": [0, 8], "memoise_2": 0, "rcurl_1": 0, "98": 0, "ggtree_3": 0, "s4arrays_1": 0, "htmltools_0": 0, "sparsearray_1": 0, "gridgraphics_0": 0, "plyr_1": 0, "cachem_1": 0, "igraph_2": 0, "lifecycle_1": 0, "pkgconfig_2": 0, "23": 0, "rsvd_1": 0, "matrix_1": 0, "r6_2": 0, "fastmap_1": 0, "27": [0, 8], "gson_0": 0, "genomeinfodbdata_1": 0, "29": 0, "matrixgenerics_1": 0, "digest_0": 0, "34": 0, "aplot_0": 0, "colorspace_2": 0, "33": 0, "patchwork_1": 0, "annotationdbi_1": 0, "35": 0, "s4vectors_0": 0, "40": 0, "dqrng_0": 0, "37": 0, "irlba_2": 0, "rsqlite_2": 0, "39": 0, "beachmat_2": 0, "fansi_1": 0, "41": 0, "timechange_0": 0, "abind_1": 0, "43": 0, "httr_1": 0, "polyclip_1": 0, "45": 0, "compiler_4": 0, "bit64_4": 0, "47": 0, "withr_3": 0, "biocparallel_1": 0, "36": 0, "49": 0, "viridis_0": 0, "dbi_1": 0, "51": 0, "ggforce_0": 0, "mass_7": 0, "53": 0, "delayedarray_0": 0, "28": [0, 5], "hdo": 0, "db_0": 0, "99": 0, "55": 0, "tools_4": 0, "ape_5": 0, "57": 0, "scatterpie_0": 0, "glue_1": 0, "59": 0, "nlme_3": 0, "164": 0, "gosemsim_2": 0, "61": 0, "grid_4": 0, "shadowtext_0": 0, "63": 0, "reshape2_1": 0, "fgsea_1": 0, "65": 0, "generics_0": 0, "gtable_0": 0, "67": 0, "tzdb_0": 0, "preprocesscore_1": 0, "69": 0, "table_1": 0, "hms_1": 0, "71": 0, "scaledmatrix_1": 0, "biocsingular_1": 0, "73": 0, "tidygraph_1": 0, "utf8_1": 0, "75": 0, "xvector_0": 0, "42": 0, "pillar_1": 0, "77": 0, "utils_0": 0, "splines_4": 0, "79": 0, "tweenr_2": 0, "treeio_1": 0, "26": [0, 8], "81": 0, "lattice_0": 0, "bit_4": 0, "83": 0, "tidyselect_1": 0, "db_3": 0, "85": 0, "biostrings_2": 0, "70": 0, "knitr_1": 0, "87": 0, "gridextra_2": 0, "iranges_2": 0, "89": 0, "stats4_4": 0, "xfun_0": 0, "91": 0, "graphlayouts_1": 0, "statmod_1": 0, "93": 0, "matrixstats_1": 0, "stringi_1": 0, "95": 0, "lazyeval_0": 0, "ggfun_0": 0, "97": 0, "yaml_2": 0, "evaluate_0": 0, "codetools_0": 0, "ggraph_2": 0, "101": 0, "qvalue_2": 0, "biocmanager_1": 0, "103": 0, "ggplotify_0": 0, "cli_3": 0, "105": 0, "affyio_1": 0, "72": 0, "munsell_0": 0, "107": 0, "rcpp_1": 0, "genomeinfodb_1": 0, "38": 0, "109": 0, "png_0": 0, "parallel_4": 0, "111": 0, "blob_1": 0, "dose_3": 0, "113": [0, 8], "sparsematrixstats_1": 0, "bitops_1": 0, "115": 0, "viridislite_0": 0, "tidytree_0": 0, "117": 0, "scales_1": 0, "crayon_1": 0, "119": 0, "rlang_1": 0, "cowplot_1": 0, "121": 0, "fastmatch_1": 0, "keggrest_1": 0, "http": [0, 1, 4, 5, 8, 9], "releas": [0, 4, 8, 9], "bioc": 0, "inst": 0, "doc": [0, 4, 5], "usersguid": 0, "pdf": 0, "biomed": 0, "knowledg": 0, "mine": 0, "book": [0, 5, 6], "html": [0, 1, 4, 8], "davetang": 0, "muse": 0, "2018": 0, "due": [1, 9], "coupl": 1, "along": 1, "summar": [1, 6], "includ": 1, "miniconda3": [1, 6], "nextflow": 1, "singular": [1, 2], "instal": [1, 4, 6, 9], "fasta": 1, "compress": 1, "mn908947": [1, 2], "zip": 1, "contain": [1, 3, 6], "wuhan": 1, "directori": [1, 5, 8, 9], "activ": [1, 2, 4, 8, 9], "mooc": [1, 8, 9], "environ": [1, 9], "conda": [1, 2, 6, 8, 9], "data": [1, 3, 4, 6], "fastq": [1, 8], "dump": 1, "split": [1, 8], "err5743893": 1, "folder": [1, 5, 7, 9], "store": [1, 7, 9], "report": 1, "mkdir": [1, 5, 8], "qc_report": [1, 8], "fastqc": 1, "we": [1, 8], "err5743893_1": 1, "err5743893_2": 1, "outdir": [1, 2, 8, 9], "multiqc": 1, "open": [1, 4, 7, 8, 9], "multiqc_report": 1, "web": 1, "browser": 1, "output": [1, 8], "genom": [1, 2, 6, 8, 9], "fai": 1, "sam": 1, "binari": 1, "bam": [1, 8], "quicker": 1, "downstream": 1, "bai": 1, "integr": 1, "viewer": 1, "igv": 1, "bwa": 1, "sinc": 1, "produc": 1, "samtool": [1, 8], "faidx": 1, "mem": 1, "cd": [1, 5, 9], "l": [1, 8], "lhrt": 1, "b": [1, 5], "thread": 1, "spec": [1, 2], "confirm": [1, 5, 8, 9], "32": [1, 8], "softwar": [1, 9], "refseq": 1, "track": 1, "sampleseq": 1, "freebay": 1, "vcf": [1, 8], "f": [1, 4, 8], "It": [1, 4, 8], "good": 1, "practiv": 1, "less": 1, "space": [1, 4, 9], "uncompress": 1, "them": 1, "bgzip": [1, 8], "tabix": [1, 8], "gz": [1, 8], "exampl": [1, 8, 9], "quickli": 1, "retriev": 1, "bcftool": [1, 8], "err5743893_stat": 1, "output_snp": 1, "z": 1, "snp": 1, "indel": 1, "queri": 1, "type": [1, 5, 9], "uniq": 1, "wellcom": [1, 3, 6, 9], "connect": [1, 3, 5, 6, 9], "scienc": [1, 3, 6, 9], "cours": [1, 6, 9], "futurelearn": [1, 9], "bioinformat": [1, 8, 9], "biologist": [1, 6, 9], "analys": [1, 9], "cdcgov": 1, "sar": 1, "cov": 1, "2": [1, 5], "www": [1, 5, 8], "babraham": 1, "ac": [1, 8], "uk": [1, 8], "project": [1, 4, 5, 8], "app": [1, 8], "nlm": 1, "nih": 1, "gov": 1, "pmc": 1, "articl": 1, "pmc5860075": 1, "dnatech": 1, "genomecent": 1, "ucdavi": 1, "edu": 1, "faq": 1, "when": [1, 4, 5], "trim": [1, 8], "illumina": [1, 2], "pmc3137218": 1, "nf": [2, 8, 9], "core": [2, 8, 9], "profil": [2, 8, 9], "max_memori": 2, "gb": [2, 8], "max_cpu": 2, "4": [2, 9], "whatev": [2, 9], "samplesheet": [2, 8], "csv": [2, 8], "protocol": 2, "amplicon": 2, "3": [2, 4], "primer_set": 2, "artic": 2, "primer_set_vers": 2, "skip_kraken2": 2, "skip_assembli": 2, "skip_pangolin": 2, "skip_nextclad": 2, "chapter": 3, "regular": 3, "workflow": 3, "summari": [3, 5, 6, 8], "channel": 4, "repositori": 4, "forg": 4, "curat": 4, "commun": 4, "driven": 4, "env": [4, 9], "own": 4, "content": 4, "interact": 4, "ensur": [4, 8], "reproduc": 4, "reusabl": 4, "program": 4, "mai": [4, 8], "intend": 4, "declar": 4, "becaus": 4, "compat": 4, "practic": 4, "yaml": 4, "within": 4, "yml": [4, 9], "onc": 4, "purpos": [4, 6], "specifi": [4, 5], "new": [4, 5, 7, 9], "section": 4, "vi": 4, "geoproj": 4, "python": 4, "6": 4, "now": [4, 8, 9], "diff": 4, "highest": 4, "seri": 4, "8": 4, "option": [4, 5, 7], "prevent": 4, "io": [4, 9], "en": [4, 5], "user": [4, 5, 9], "guid": 4, "concept": 4, "geohackweek": 4, "github": [4, 9], "datashar": 4, "01": 4, "tutori": [4, 5], "edcarp": 4, "introduct": 4, "scientist": 4, "newproject": 5, "root": [5, 9], "branch": 5, "right": 5, "awai": 5, "init": 5, "push": 5, "easier": 5, "otherwis": 5, "start": [5, 8, 9], "your": 5, "command": [5, 9], "stage": 5, "commit": 5, "ipynb": 5, "exclus": 5, "certain": 5, "gitignor": 5, "never": 5, "touch": 5, "nano": 5, "except": 5, "jupyt": [5, 6], "notebook": 5, "would": 5, "ipynb_checkpoint": 5, "atlassian": 5, "statu": 5, "rememb": 5, "about": 5, "cach": 5, "rm": [5, 9], "r": [5, 6, 8], "safe": 5, "choic": 5, "reset": 5, "re": [5, 9], "call": [5, 9], "dai": 5, "gener": [5, 7, 8], "pat": 5, "person": 5, "access": [5, 9], "token": 5, "develop": 5, "somewher": 5, "without": [5, 8], "readm": 5, "md": 5, "copi": 5, "url": [5, 9], "past": 5, "let": 5, "master": 5, "ask": 5, "password": 5, "scm": 5, "com": [5, 8, 9], "v2": 5, "migrat": 5, "code": 5, "host": 5, "datacamp": 5, "pull": 5, "One": 6, "learn": 6, "self": [6, 9], "review": 6, "explan": 6, "undertaken": 6, "site": [6, 8, 9], "wsl": [6, 9], "remind": 6, "usag": 6, "git": 6, "basic": [6, 9], "bash": [6, 9], "heart": 6, "stack": 6, "troubleshoot": 6, "incompat": 6, "error": [7, 9], "solv": 7, "config": [7, 8], "unhash": 7, "close": [7, 9], "consol": 7, "trick": 7, "2025": 8, "09": 8, "wsl2": [8, 9], "docker": 8, "miniconda": [8, 9], "tumor": 8, "nascent": 8, "adjac": 8, "tissu": 8, "pbmc": 8, "ensembl": 8, "soft": 8, "mask": 8, "primari": 8, "assembli": 8, "gatk4": 8, "truncat": 8, "seqtk": 8, "gzip": 8, "s100": 8, "100000": 8, "sub1": 8, "sub2": 8, "qualiti": 8, "befor": 8, "Then": 8, "skip": 8, "those": 8, "_1": 8, "_2": 8, "ourdir": 8, "associ": 8, "rnaseq": [8, 9], "script": 8, "goe": 8, "path_to_project": 8, "nsclc_cohort": 8, "setwd": 8, "str_sub": 8, "fastq_1": 8, "keep": 8, "str_detect": 8, "fastq_2": 8, "stranded": 8, "auto": 8, "write_csv": 8, "termin": [8, 9], "sudo": [8, 9], "mnt": 8, "homo_sapien": 8, "grch38": 8, "dna_sm": 8, "primary_assembli": 8, "fa": 8, "gtf": 8, "star_index": 8, "star_refer": 8, "email": 8, "mamuszcz": 8, "gmail": 8, "align": 8, "star_salmon": 8, "stringtie_ignore_gtf": 8, "stringti": 8, "save_align_interm": 8, "skip_fastqc": 8, "extra_star_align_arg": 8, "outtmpdir": 8, "star_temp": 8, "look": 8, "resourcelimit": 8, "cpu": 8, "leav": 8, "memori": [8, 9], "withnam": 8, "star_genomegener": 8, "wanna": 8, "ext": 8, "genomechrbinnbit": 8, "genomesaspars": 8, "join": 8, "gffcompar": 8, "fresh": 8, "abund": 8, "gatk": 8, "createsequencedictionari": 8, "With": 8, "must": 8, "splt": 8, "cigar": 8, "splice": 8, "junction": 8, "isn": 8, "awar": 8, "splitncigarread": 8, "samplea": 8, "markdup": 8, "sampleb": 8, "dont": 8, "clip": 8, "sample_somatic_vari": 8, "score": 8, "genotyp": 8, "filtermutectcal": 8, "sample_somatic_variants_filt": 8, "benign": 8, "haplotypecal": 8, "xmx26g": 8, "erc": 8, "gvcf": 8, "joint": 8, "healthi": 8, "intron": 8, "common": 8, "snv": 8, "combinegvcf": 8, "sample_combin": 8, "genotypegvcf": 8, "sample_germline_vari": 8, "pass": 8, "grep": 8, "wc": 8, "record": 8, "sample_somatic_variants_filt2": 8, "snpeff": 8, "estim": 8, "effect": 8, "p14": 8, "cvsstat": 8, "sample_vc_summari": 8, "sample_somatic_variants_ann": 8, "snpsift": 8, "clinvar": 8, "sample_somatic_variants_ann_clinvar": 8, "info": 8, "ann": 8, "missens": 8, "stop_gain": 8, "stop_lost": 8, "start_lost": 8, "sample_somatic_variants_ann_n": 8, "intersect": 8, "share": 8, "isec": 8, "sample_shared_vari": 8, "identifi": 8, "none": 8, "sample_somatic_only_vari": 8, "sample_germline_only_vari": 8, "dp": 8, "polymorph": 8, "fastaalternatereferencemak": 8, "acc": 8, "perplex": 8, "either": 8, "onlin": 8, "ebi": 8, "jdispatch": 8, "st": 8, "emboss_transeq": 8, "sms2": 8, "offlin": 8, "biostr": 8, "biopython": 8, "bio": 8, "alphabet": 8, "iupac": 8, "modul": 8, "offici": 9, "instruct": 9, "short": 9, "enabl": 9, "virtual": 9, "machin": 9, "ubuntu": 9, "microsoft": 9, "page": 9, "wslconfig": 9, "usernam": 9, "kernelcommandlin": 9, "vsyscal": 9, "emul": 9, "u": 9, "processor": 9, "insid": 9, "shell": 9, "java": 9, "apt": 9, "jre": 9, "curl": 9, "fssl": 9, "mv": 9, "bin": 9, "reopen": 9, "help": 9, "ipython": 9, "1": 9, "875bb003884f": 9, "syntaxerror": 9, "invalid": 9, "syntax": 9, "pipelin": 9, "were": 9, "configur": 9, "correctli": 9, "control": 9, "disk": 9, "bigger": 9, "over": 9, "anaconda": 9, "wget": 9, "repo": 9, "continuum": 9, "sh": 9, "cannot": 9, "export": 9, "unzip": 9, "transfer": 9, "desir": 9, "support": 9, "recommend": 9, "instead": 9, "unlik": 9, "doesn": 9, "admin": 9, "04": 9, "runc": 9, "cryptsetup": 9, "deb": 9, "sylab": 9, "v3": 9, "ce_3": 9, "jammy_amd64": 9, "dpkg": 9, "jammi": 9, "tojamm": 9, "focal_amd64": 9, "fix": 9, "broken": 9, "uidmap": 9, "try": 9, "ce": 9}, "objects": {}, "objtypes": {}, "objnames": {}, "titleterms": {"r": 0, "stack": 0, "omic": 0, "data": [0, 8], "scope": 0, "1": [0, 1, 8], "affi": 0, "2": [0, 2, 8], "limma": 0, "3": [0, 8], "clusterprofil": 0, "4": [0, 8], "pheatmap": 0, "5": [0, 8], "pcatool": 0, "6": [0, 8], "eulerr": 0, "7": [0, 8], "corrplot": 0, "8": [0, 8], "ggpubr": 0, "patchwork": 0, "session": 0, "info": [0, 1, 5], "more": [0, 1, 5], "sourc": [0, 1, 4, 5, 9], "week": [1, 2], "setup": [1, 9], "sequenc": [1, 8], "qc": 1, "map": [1, 8], "variant": [1, 8], "call": [1, 8], "qualiti": 1, "control": 1, "sampl": [1, 8], "refer": [1, 8], "workflow": 2, "nextflow": [2, 8, 9], "run": 2, "viralrecon": 2, "pipelin": [2, 8], "cours": 3, "note": [3, 4, 8], "bioinformat": [3, 6], "biologist": 3, "conda": 4, "structur": 4, "creat": [4, 8], "environ": [4, 8], "us": 4, "command": 4, "about": 4, "packag": 4, "updat": 4, "git": 5, "github": 5, "instal": 5, "window": [5, 9], "wsl": [5, 7], "repo": 5, "guid": 5, "initi": 5, "local": 5, "work": 5, "track": 5, "cheatsheet": 6, "incompat": 7, "solut": 7, "jupyt": 7, "lab": 7, "server": 7, "start": 7, "proteogenom": 8, "neoantigen": 8, "discoveri": 8, "overview": 8, "comput": 8, "set": 8, "up": 8, "input": 8, "tool": 8, "step": 8, "fastqc": 8, "multiqc": 8, "sheet": 8, "prepar": 8, "rna": 8, "seq": 8, "option": 8, "star": 8, "evalu": 8, "new": 8, "transcript": 8, "fasta": 8, "dictionari": 8, "somat": 8, "mutect2": 8, "germlin": 8, "annot": 8, "filter": 8, "non": 8, "synonym": 8, "i": 8, "silent": 8, "result": 8, "chang": 8, "aa": 8, "find": 8, "mutat": 8, "which": 8, "were": 8, "found": 8, "AND": 8, "reason": 8, "read": 8, "depth": 8, "incorpor": 8, "nucleotid": 8, "9": 8, "translat": 8, "frame": 8, "three": 8, "forward": 8, "revers": 8, "system": 9, "subsystem": 9, "linux": 9, "docker": 9, "miniconda3": 9, "singular": 9}, "envversion": {"sphinx.domains.c": 2, "sphinx.domains.changeset": 1, "sphinx.domains.citation": 1, "sphinx.domains.cpp": 8, "sphinx.domains.index": 1, "sphinx.domains.javascript": 2, "sphinx.domains.math": 2, "sphinx.domains.python": 3, "sphinx.domains.rst": 2, "sphinx.domains.std": 2, "sphinx.ext.intersphinx": 1, "sphinxcontrib.bibtex": 9, "sphinx": 57}, "alltitles": {"R stack for -omics data": [[0, "r-stack-for-omics-data"]], "Scope": [[0, "scope"]], "1. affy": [[0, "affy"]], "2. limma": [[0, "limma"]], "3. clusterProfiler": [[0, "clusterprofiler"]], "4. pheatmap": [[0, "pheatmap"]], "5. PCAtools": [[0, "pcatools"]], "6. eulerr": [[0, "eulerr"]], "7. corrplot": [[0, "corrplot"]], "8. ggpubr & patchwork": [[0, "ggpubr-patchwork"]], "Session info": [[0, "session-info"]], "More sources": [[0, "more-sources"]], "Week 1 - setup, sequencing QC, mapping, variant calling": [[1, "week-1-setup-sequencing-qc-mapping-variant-calling"]], "Setup": [[1, "setup"]], "Sequencing Quality Control": [[1, "sequencing-quality-control"]], "Mapping sample sequence to reference": [[1, "mapping-sample-sequence-to-reference"]], "Variant calling": [[1, "variant-calling"]], "Sources": [[1, "sources"], [5, "sources"], [9, "sources"]], "More info": [[1, "more-info"], [5, "more-info"]], "Week 2 - workflows with Nextflow": [[2, "week-2-workflows-with-nextflow"]], "Running \u2018viralrecon\u2019 pipeline": [[2, "running-viralrecon-pipeline"]], "Course notes: Bioinformatics for Biologists": [[3, "course-notes-bioinformatics-for-biologists"]], "Conda": [[4, "conda"]], "Structure": [[4, "structure"]], "Creating environments": [[4, "creating-environments"]], "Useful commands": [[4, "useful-commands"]], "Notes about package updates": [[4, "notes-about-package-updates"]], "Sources:": [[4, "sources"]], "Git & GitHub": [[5, "git-github"]], "Git installation - Windows/WSL": [[5, "git-installation-windows-wsl"]], "Git repo guide": [[5, "git-repo-guide"]], "Initializing a local repo": [[5, "initializing-a-local-repo"]], "Working with repo tracking": [[5, "working-with-repo-tracking"]], "GitHub guide": [[5, "github-guide"]], "Bioinformatics cheatsheets": [[6, "bioinformatics-cheatsheets"]], "WSL incompatibility solutions": [[7, "wsl-incompatibility-solutions"]], "Jupyter Lab server not starting": [[7, "jupyter-lab-server-not-starting"]], "Proteogenomics for neoantigen discovery": [[8, "proteogenomics-for-neoantigen-discovery"]], "Overview": [[8, "overview"]], "Computational set-up": [[8, "computational-set-up"]], "Data input:": [[8, "data-input"]], "Tools and environments:": [[8, "tools-and-environments"]], "Notes:": [[8, "notes"]], "Step by step:": [[8, "step-by-step"]], "1. FASTQC and MULTIQC": [[8, "fastqc-and-multiqc"]], "2. Sample sheet preparation": [[8, "sample-sheet-preparation"]], "3. RNA-seq Nextflow pipeline with optional reference mapping with STAR": [[8, "rna-seq-nextflow-pipeline-with-optional-reference-mapping-with-star"]], "Evaluate new transcripts:": [[8, "evaluate-new-transcripts"]], "4. Create fasta dictionary": [[8, "create-fasta-dictionary"]], "5. Variant calling": [[8, "variant-calling"]], "Somatic variants call - Mutect2": [[8, "somatic-variants-call-mutect2"]], "Germline variant calling": [[8, "germline-variant-calling"]], "Variant annotation and filtering": [[8, "variant-annotation-and-filtering"]], "6. Filter for non-synonymous variants (i.a. non-silent, resulting in a change in AA sequence)": [[8, "filter-for-non-synonymous-variants-i-a-non-silent-resulting-in-a-change-in-aa-sequence"]], "7. Find mutations which were found as germline AND somatic": [[8, "find-mutations-which-were-found-as-germline-and-somatic"]], "Reasonable read depth for variants:": [[8, "reasonable-read-depth-for-variants"]], "8. Incorporate variants into nucleotide FASTA reference": [[8, "incorporate-variants-into-nucleotide-fasta-reference"]], "9. Translate nucleotide FASTA with variants to FASTA AA sequence with 6 frames (three forward and three reverse frames)": [[8, "translate-nucleotide-fasta-with-variants-to-fasta-aa-sequence-with-6-frames-three-forward-and-three-reverse-frames"]], "System setup": [[9, "system-setup"]], "Windows Subsystem for Linux": [[9, "windows-subsystem-for-linux"]], "Docker": [[9, "docker"]], "Nextflow": [[9, "nextflow"]], "Miniconda3": [[9, "miniconda3"]], "*Singularity": [[9, "singularity"]]}, "indexentries": {}})